{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "50bbf1de-f394-4c48-afb4-a5a42263f6c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Foldseek enables fast and sensitive comparisons of large structure sets. It reaches sensitivities similar to state-of-the-art structural aligners while being at least 20,000 times faster.\n",
      "\n",
      "Please cite:\n",
      "van Kempen, M., Kim, S.S., Tumescheit, C., Mirdita, M., Lee, J., Gilchrist, C.L.M., Söding, J., and Steinegger, M. Fast and accurate protein structure search with Foldseek. Nature Biotechnology, doi:10.1038/s41587-023-01773-0 (2023)\n",
      "\n",
      "foldseek Version: 8.ef4e960\n",
      "© Michel van Kempen, Stephanie Kim, Charlotte Tumescheit, Milot Mirdita, Jeongjae Lee, Cameron L. M. Gilchrist, Johannes Söding, Martin Steinegger\n",
      "\n",
      "usage: foldseek <command> [<args>]\n",
      "\n",
      "Easy workflows for plain text input/output\n",
      "  easy-search       \tStructual search\n",
      "  easy-cluster      \tSlower, sensitive clustering\n",
      "  easy-rbh          \tFind reciprocal best hit\n",
      "  easy-complexsearch\tComplex level search\n",
      "\n",
      "Main workflows for database input/output\n",
      "  createdb          \tConvert PDB/mmCIF/tar[.gz]/DB files to a db\n",
      "  search            \tSensitive homology search\n",
      "  rbh               \tReciprocal best hit search\n",
      "  cluster           \tSlower, sensitive clustering\n",
      "\n",
      "Input database creation\n",
      "  databases         \tList and download databases\n",
      "  createindex       \tStore precomputed index on disk to reduce search overhead\n",
      "  createclusearchdb \tBuild a searchable cluster database allowing for faster searches\n",
      "\n",
      "Format conversion for downstream processing\n",
      "  convertalis       \tConvert alignment DB to BLAST-tab, SAM or custom format\n",
      "  convert2pdb       \tConvert a foldseek structure db to a multi model PDB file\n",
      "  createcomplexreport\tConvert complex DB to tsv format\"\n",
      "\n",
      "An extended list of all modules can be obtained by calling 'foldseek -h'.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "foldseek"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dae52caa-628d-4f21-931a-860244e21f03",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create directory tmpDir\n",
      "databases Alphafold/Swiss-Prot foldseek_DB tmpDir \n",
      "\n",
      "MMseqs Version:              \t8.ef4e960\n",
      "Tsv                          \tfalse\n",
      "Force restart with latest tmp\tfalse\n",
      "Remove temporary files       \tfalse\n",
      "Compressed                   \t0\n",
      "Threads                      \t8\n",
      "Verbosity                    \t3\n",
      "\n",
      "\n",
      "10/12 16:00:55 [\u001b[1;32mNOTICE\u001b[0m] Downloading 1 item(s)\n",
      "\u001b[35m[\u001b[0m#f252fc 1.2GiB/1.2GiB\u001b[36m(99%)\u001b[0m CN:1 DL:\u001b[32m34MiB\u001b[0m\u001b[35m]\u001b[0m\u001b[0m0m\u001b[35m]\u001b[0m\u001b[0mm\n",
      "10/12 16:01:33 [\u001b[1;32mNOTICE\u001b[0m] Download complete: tmpDir/539832902454257641/afdb_swissprot.tar.gz\n",
      "\n",
      "Download Results:\n",
      "gid   |stat|avg speed  |path/URI\n",
      "======+====+===========+=======================================================\n",
      "f252fc|\u001b[1;32mOK\u001b[0m  |    35MiB/s|tmpDir/539832902454257641/afdb_swissprot.tar.gz\n",
      "\n",
      "Status Legend:\n",
      "(OK):download completed.\n",
      "\n",
      "10/12 16:01:33 [\u001b[1;32mNOTICE\u001b[0m] Downloading 1 item(s)\n",
      "\u001b[35m[\u001b[0m#1147c2 0B/0B CN:1 DL:\u001b[32m0B\u001b[0m\u001b[35m]\u001b[0m\u001b[0m                      \n",
      "10/12 16:01:34 [\u001b[1;32mNOTICE\u001b[0m] Download complete: tmpDir/539832902454257641/version\n",
      "\n",
      "Download Results:\n",
      "gid   |stat|avg speed  |path/URI\n",
      "======+====+===========+=======================================================\n",
      "1147c2|\u001b[1;32mOK\u001b[0m  |     200B/s|tmpDir/539832902454257641/version\n",
      "\n",
      "Status Legend:\n",
      "(OK):download completed.\n",
      "afdb_swissprot\n",
      "afdb_swissprot.dbtype\n",
      "afdb_swissprot.index\n",
      "afdb_swissprot_ca\n",
      "afdb_swissprot_ca.dbtype\n",
      "afdb_swissprot_ca.index\n",
      "afdb_swissprot_ss\n",
      "afdb_swissprot_ss.dbtype\n",
      "afdb_swissprot_ss.index\n",
      "afdb_swissprot_h\n",
      "afdb_swissprot_h.dbtype\n",
      "afdb_swissprot_h.index\n",
      "afdb_swissprot_mapping\n",
      "afdb_swissprot_taxonomy\n",
      "afdb_swissprot.lookup\n",
      "mvdb tmpDir/539832902454257641/afdb_swissprot foldseek_DB \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb tmpDir/539832902454257641/afdb_swissprot_ss foldseek_DB_ss \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mvdb tmpDir/539832902454257641/afdb_swissprot_h foldseek_DB_h \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "mvdb tmpDir/539832902454257641/afdb_swissprot_ca foldseek_DB_ca \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n"
     ]
    }
   ],
   "source": [
    "# download Swiss-Prot database\n",
    "foldseek databases Alphafold/Swiss-Prot foldseek_DB tmpDir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8cb6290d-ff53-4c56-8187-81e198ad93bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MSLEQKKGADIISKILQIQNSIGKTTSPSTLKTKLSEISRKEQENARIQSKLSDLQKKKIDIDNKLLKEKQNLIKEEILERKKLEVLTKKQQKDEIEHQKKLKREIDAIKASTQYITDVSISSYNNTIPETEPEYDLFISHASEDKEDFVRPLAETLQQLGVNVWYDEFTLKVGDSLRQKIDSGLRNSKYGTVVLSTDFIKKDWTNYELDGLVAREMNGHKMILPIWHKITKNDVLDYSPNLADKVALNTSVNSIEEIAHQLADVILNR\n",
      "\u0000MTSHGAVKIAIFAVIALHSIFECLSKPQILQRTDHSTDSDWDPQMCPETCNPSKNISCSSECLCVTLGGGDETGTCFNMSGVDWLGHAQASDGHNDG\n",
      "\u0000MKVLLYIAASCLMLLALNVSAENTQQEEEDYDYGTDTCPFPVLANKTNKAKFVGCHQKCNGGDQKLTDGTACYVVERKVWDRMTPMLWYSCPLGECKNGVCEDLRKKEECRKGNGEEK\n",
      "\u0000MKVLLCIAASCLMLLALNVSAENTQQEEQDYDYGTDTCPFPVLANKTNKAKFVGCHQKCNGGDQKLTDGTACYVVERKVWDRMTPMLWYECPLGECKNGVCEDLRKKEDCRKGNGEEK\n",
      "\u0000MTSHSAVRIAIFAVIALHSIFECLSKPQILQRTDKSTDSEWDPQTCPETCIPSKNITCSDGCVCVKLGEEEEGTCFNMTGVDWLGSPSDD\n",
      "\u0000MKASFCVIASCLVVFALKGTAEDTGTEDDFDYGNTGCPFPVLGNYKSNMTKPVGCKNKCGSGYEVLNDTTPCYVIDQKVFNNMVPLRQYSKCPLGFCENGECKPNDQAEDCYKGREEQK\n",
      "\u0000MHSTIVYACLLALAVFVALHGTPLAALAENGEGTTQPDYDNSTDYYNYEDFKCTCPAPHLNNTNGTVMKPIGCYYTCNVTRCTAPDTYPCYNLTEHQAKNLTTSPTTLCAVGNCDHGICVPNGTKELCFKAPNLEE\n",
      "\u0000MHSTIAYVSLLPLALFVAMHGASTDEESEELGASTDVDYEELDANCTCPAPALTSTRNNKHYPLGCIYNCSSYNCTIPDGTPCYVLTLGEVKEHLQIGSTVPNCTCGLCRNGTCVSNGTVEECFAVEEIEET\n",
      "\u0000MARNWSFRVIFVSAMWCALLKFATLEEPKDGYDYTEGCPFVVLGNGTHAKPAGCSHLCNGAPETLDDNMECYNVTEEVAKRMTPDIPYTCWLGWCSKGECKRDNRTEVCYRGSERE\n",
      "\u0000MTRNWSFRVIFVSAMWCALLKFATLEAPKDDFEYDGGCPFVVLDNGTHVKPAGCSHLCNGAPETLDNIECYNVTEEVAKRMTPGIPYACWLGWCSKGECKRDNRTEVCYRGSEEE\n"
     ]
    }
   ],
   "source": [
    "head foldseek_DB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "501943bf-f451-45ff-8ba4-f83d34b88e3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./foldseek_DB_ss\n",
      "./foldseek_DB.dbtype\n",
      "./foldseek_DB_h.index\n",
      "./foldseek_DB_ca\n",
      "./foldseek_DB_taxonomy\n",
      "./foldseek_DB_ca.index\n",
      "./foldseek_DB\n",
      "./foldseek_DB_mapping\n",
      "./foldseek_DB.index\n",
      "./foldseek_DB.version\n",
      "./foldseek_DB.lookup\n",
      "./foldseek_DB_h\n",
      "./foldseek_DB_h.dbtype\n",
      "./foldseek_DB_ss.dbtype\n",
      "./foldseek_DB_ca.dbtype\n",
      "./foldseek_DB_ss.index\n"
     ]
    }
   ],
   "source": [
    "find . -type f -name \"foldseek_DB*\" \n",
    "\n",
    "# foldseek_DB contains aa info\n",
    "# foldseek_DB_h.index contains header info\n",
    "# foldseek_DB_ss.index contains 3D sequence info\n",
    "# foldseek_DB_ca.index contains c-alpha coordinate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83441965-06c4-42c8-aea3-a4ecbdc94f85",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "a68a9e19-3e5c-47e5-87bf-dad387e1e56f",
   "metadata": {},
   "source": [
    "# Clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "107a5921-2382-4919-932d-e821746058f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir foldseek_cluster_0.80"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3f7a5c79-595e-4ced-bb97-63d59ceb0fd2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create directory tmp\n",
      "cluster foldseek_DB foldseek_cluster_0.80/foldseek_cluster_0.80 tmp --min-seq-id 0.80 \n",
      "\n",
      "MMseqs Version:                     \t8.ef4e960\n",
      "Substitution matrix                 \taa:3di.out,nucl:3di.out\n",
      "Seed substitution matrix            \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t1000\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t0\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t1\n",
      "Minimum diagonal score              \t30\n",
      "Selected taxa                       \t\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "TMscore threshold                   \t0\n",
      "LDDT threshold                      \t0\n",
      "Sort by structure bit score         \t0\n",
      "Alignment type                      \t2\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "E-value threshold                   \t0.01\n",
      "Seq. id. threshold                  \t0.8\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Gap open cost                       \taa:10,nucl:10\n",
      "Gap extension cost                  \taa:1,nucl:1\n",
      "TMalign hit order                   \t0\n",
      "TMalign fast                        \t1\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t300\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "kmermatcher foldseek_DB_ss tmp/2822590051813142283/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.8 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher foldseek_DB_ss tmp/2822590051813142283/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.8 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 542378 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A F) (C V) (D B) (E Z) (G H) (I M T) (K W) (L J) (N R S) (P) (Q) (Y) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 542.38K 2s 339ms    \n",
      "Sort kmer 0h 0m 6s 119ms\n",
      "Sort by rep. sequence 0h 0m 1s 808ms\n",
      "Time for fill: 0h 0m 1s 477ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 14s 573ms\n",
      "structurerescorediagonal foldseek_DB foldseek_DB tmp/2822590051813142283/pref tmp/2822590051813142283/pref_rescore1 --tmscore-threshold 0 --lddt-threshold 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.8 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=========>                                                       ] 15.00% 81.38K eta 36s        \n"
     ]
    }
   ],
   "source": [
    "# took around 1.5hr to run the command ?!\n",
    "foldseek cluster foldseek_DB foldseek_cluster_0.80/foldseek_cluster_0.80 tmp --min-seq-id 0.80"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2117d2fe-bf8e-4224-918a-b6acc9bbca45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createtsv foldseek_DB foldseek_DB foldseek_cluster_0.80/foldseek_cluster_0.80 foldseek_cluster_0.80/foldseek_cluster_0.80.tsv \n",
      "\n",
      "MMseqs Version:                 \t8.ef4e960\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t8\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to foldseek_cluster_0.80.tsv: 0h 0m 0s 117ms\n",
      "Time for processing: 0h 0m 0s 402ms\n"
     ]
    }
   ],
   "source": [
    "foldseek createtsv foldseek_DB foldseek_DB foldseek_cluster_0.80/foldseek_cluster_0.80 foldseek_cluster_0.80/foldseek_cluster_0.80.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8576e484-4109-434e-8dd6-a2672def3ae8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AF-P82289-F1-model_v4\tAF-P82289-F1-model_v4\n",
      "AF-P02020-F1-model_v4\tAF-P02020-F1-model_v4\n",
      "AF-P82978-F1-model_v4\tAF-P82978-F1-model_v4\n",
      "AF-P02138-F1-model_v4\tAF-P02138-F1-model_v4\n",
      "AF-O82712-F1-model_v4\tAF-O82712-F1-model_v4\n",
      "AF-O82713-F1-model_v4\tAF-O82713-F1-model_v4\n",
      "AF-Q75VA5-F1-model_v4\tAF-Q75VA5-F1-model_v4\n",
      "AF-Q75VA5-F1-model_v4\tAF-Q75VA9-F1-model_v4\n",
      "AF-Q75VA5-F1-model_v4\tAF-Q75VB4-F1-model_v4\n",
      "AF-Q75VA5-F1-model_v4\tAF-Q67BD8-F1-model_v4\n"
     ]
    }
   ],
   "source": [
    "head foldseek_cluster_0.80.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "838f67ed-142e-41d6-856e-487d28294b6f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ee2e0884-7f4a-4b16-ac40-064668f3c2b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster foldseek_DB foldseek_cluster_0.85/foldseek_cluster_0.85 tmp --min-seq-id 0.85 \n",
      "\n",
      "MMseqs Version:                     \t8.ef4e960\n",
      "Substitution matrix                 \taa:3di.out,nucl:3di.out\n",
      "Seed substitution matrix            \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t1000\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t0\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t1\n",
      "Minimum diagonal score              \t30\n",
      "Selected taxa                       \t\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "TMscore threshold                   \t0\n",
      "LDDT threshold                      \t0\n",
      "Sort by structure bit score         \t0\n",
      "Alignment type                      \t2\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "E-value threshold                   \t0.01\n",
      "Seq. id. threshold                  \t0.85\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Gap open cost                       \taa:10,nucl:10\n",
      "Gap extension cost                  \taa:1,nucl:1\n",
      "TMalign hit order                   \t0\n",
      "TMalign fast                        \t1\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t300\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "structurealign foldseek_DB foldseek_DB tmp/9951140192641727792/pref_filter2 tmp/9951140192641727792/aln.linclust --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.85 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 327.61K 9m 32s 311ms    \n",
      "Time for merging to aln.linclust: 0h 0m 0s 160ms\n",
      "Time for processing: 0h 9m 32s 726ms\n",
      "createsubdb tmp/9951140192641727792/order_redundancy foldseek_DB tmp/9951140192641727792/pre_clustered_seqs -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pre_clustered_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 147ms\n",
      "clust tmp/9951140192641727792/pre_clustered_seqs tmp/9951140192641727792/aln.linclust tmp/9951140192641727792/clust.linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 327.61K 0s 253ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 15148 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 327.61K 0s 250ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 327.61K 0s 19ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 652ms\n",
      "Total time: 0h 0m 0s 761ms\n",
      "\n",
      "Size of the sequence database: 327609\n",
      "Size of the alignment database: 327609\n",
      "Number of clusters: 315090\n",
      "\n",
      "Writing results 0h 0m 0s 177ms\n",
      "Time for merging to clust.linclust: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 63ms\n",
      "mergeclusters foldseek_DB tmp/9951140192641727792/clu_redundancy tmp/9951140192641727792/pre_clust tmp/9951140192641727792/clust.linclust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 327.61K 0s 174ms    \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 315.09K 0s 390ms    \n",
      "Write merged clustering\n",
      "[============================>                                    ] 44.05% 238.89K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 315.09K 0s 991ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 315.09K 1s 107ms    \n",
      "Index statistics\n",
      "Entries:          13414118\n",
      "DB size:          565 MB\n",
      "Avg k-mer size:   0.209596\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t4871\n",
      "    AEEEDD\t3010\n",
      "    SVSHYE\t2898\n",
      "    VVHYEY\t2648\n",
      "    IWWWDP\t2319\n",
      "    YEEQVS\t2175\n",
      "    EEEALL\t1884\n",
      "    EEEEDP\t1835\n",
      "    KKKFPV\t1763\n",
      "    EEEAVV\t1683\n",
      "Time for index table init: 0h 0m 2s 701ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 315090\n",
      "Target db start 1 to 315090\n",
      "[=================================================================] 100.00% 315.09K 41s 540ms    \n",
      "\n",
      "2.788277 k-mers per position\n",
      "28727 DB matches per sequence\n",
      "25 overflows\n",
      "68 sequences passed prefiltering per query sequence\n",
      "100 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 134ms\n",
      "Time for processing: 0h 0m 45s 492ms\n",
      "structurealign tmp/9951140192641727792/input_step_redundancy tmp/9951140192641727792/input_step_redundancy tmp/9951140192641727792/pref_step0 tmp/9951140192641727792/aln_step0 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.85 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 315.09K 20m 27s 151ms    \n",
      "Time for merging to aln_step0: 0h 0m 0s 149ms\n",
      "Time for processing: 0h 20m 28s 220ms\n",
      "clust tmp/9951140192641727792/input_step_redundancy tmp/9951140192641727792/aln_step0 tmp/9951140192641727792/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 315.09K 0s 266ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 9824 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 315.09K 0s 260ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 315.09K 0s 140ms    \n",
      "\n",
      "Time for read in: 0h 0m 0s 754ms\n",
      "Total time: 0h 0m 0s 799ms\n",
      "\n",
      "Size of the sequence database: 315090\n",
      "Size of the alignment database: 315090\n",
      "Number of clusters: 300064\n",
      "\n",
      "Writing results 0h 0m 0s 126ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 27ms\n",
      "createsubdb tmp/9951140192641727792/clu_step0 tmp/9951140192641727792/input_step_redundancy_ss tmp/9951140192641727792/input_step1_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 118ms\n",
      "createsubdb tmp/9951140192641727792/clu_step0 tmp/9951140192641727792/input_step_redundancy_ca tmp/9951140192641727792/input_step1_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 93ms\n",
      "createsubdb tmp/9951140192641727792/clu_step0 tmp/9951140192641727792/input_step_redundancy tmp/9951140192641727792/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 78ms\n",
      "prefilter tmp/9951140192641727792/input_step1_ss tmp/9951140192641727792/input_step1_ss tmp/9951140192641727792/pref_step1 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 200 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 300064 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 300064 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 300.06K 1s 68ms     \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 300.06K 1s 251ms    \n",
      "Index statistics\n",
      "Entries:          12998893\n",
      "DB size:          562 MB\n",
      "Avg k-mer size:   0.203108\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t4346\n",
      "    AEEEDD\t2911\n",
      "    SVSHYE\t2807\n",
      "    VVHYEY\t2591\n",
      "    IWWWDP\t2259\n",
      "    YEEQVS\t2136\n",
      "    EEEALL\t1835\n",
      "    EEEEDP\t1791\n",
      "    AEEEPL\t1690\n",
      "    EEEAVV\t1616\n",
      "Time for index table init: 0h 0m 3s 32ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 300064\n",
      "Target db start 1 to 300064\n",
      "[=================================================================] 100.00% 300.06K 50s 496ms    \n",
      "\n",
      "1.037717 k-mers per position\n",
      "13351 DB matches per sequence\n",
      "0 overflows\n",
      "116 sequences passed prefiltering per query sequence\n",
      "200 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 156ms\n",
      "Time for processing: 0h 0m 55s 8ms\n",
      "structurealign tmp/9951140192641727792/input_step1 tmp/9951140192641727792/input_step1 tmp/9951140192641727792/pref_step1 tmp/9951140192641727792/aln_step1 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.85 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 300.06K 30m 15s 815ms    \n",
      "Time for merging to aln_step1: 0h 0m 0s 196ms\n",
      "Time for processing: 0h 30m 16s 423ms\n",
      "clust tmp/9951140192641727792/input_step1 tmp/9951140192641727792/aln_step1 tmp/9951140192641727792/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 300.06K 0s 416ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 300.06K 0s 352ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 300.06K 0s 27ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 946ms\n",
      "Total time: 0h 0m 1s 75ms\n",
      "\n",
      "Size of the sequence database: 300064\n",
      "Size of the alignment database: 300064\n",
      "Number of clusters: 299878\n",
      "\n",
      "Writing results 0h 0m 0s 199ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 542ms\n",
      "createsubdb tmp/9951140192641727792/clu_step1 tmp/9951140192641727792/input_step1_ss tmp/9951140192641727792/input_step2_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 143ms\n",
      "createsubdb tmp/9951140192641727792/clu_step1 tmp/9951140192641727792/input_step1_ca tmp/9951140192641727792/input_step2_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 201ms\n",
      "createsubdb tmp/9951140192641727792/clu_step1 tmp/9951140192641727792/input_step1 tmp/9951140192641727792/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 105ms\n",
      "prefilter tmp/9951140192641727792/input_step2_ss tmp/9951140192641727792/input_step2_ss tmp/9951140192641727792/pref_step2 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 299878 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 299878 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 299.88K 1s 303ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 299.88K 1s 510ms    \n",
      "Index statistics\n",
      "Entries:          12996196\n",
      "DB size:          562 MB\n",
      "Avg k-mer size:   0.203066\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t4346\n",
      "    AEEEDD\t2909\n",
      "    SVSHYE\t2807\n",
      "    VVHYEY\t2591\n",
      "    IWWWDP\t2259\n",
      "    YEEQVS\t2136\n",
      "    EEEALL\t1835\n",
      "    EEEEDP\t1791\n",
      "    AEEEPL\t1690\n",
      "    EEEAVV\t1616\n",
      "Time for index table init: 0h 0m 3s 710ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 299878\n",
      "Target db start 1 to 299878\n",
      "[=================================================================] 100.00% 299.88K 58s 216ms     \n",
      "\n",
      "1.037697 k-mers per position\n",
      "13352 DB matches per sequence\n",
      "0 overflows\n",
      "355 sequences passed prefiltering per query sequence\n",
      "198 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 149ms\n",
      "Time for processing: 0h 1m 3s 862ms\n",
      "structurealign tmp/9951140192641727792/input_step2 tmp/9951140192641727792/input_step2 tmp/9951140192641727792/pref_step2 tmp/9951140192641727792/aln_step2 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.85 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 299.88K 55m 55s 240ms    \n",
      "Time for merging to aln_step2: 0h 0m 0s 362ms\n",
      "Time for processing: 0h 55m 58s 333ms\n",
      "clust tmp/9951140192641727792/input_step2 tmp/9951140192641727792/aln_step2 tmp/9951140192641727792/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 299.88K 0s 401ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 299.88K 0s 393ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 299.88K 0s 22ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 914ms\n",
      "Total time: 0h 0m 1s 32ms\n",
      "\n",
      "Size of the sequence database: 299878\n",
      "Size of the alignment database: 299878\n",
      "Number of clusters: 299876\n",
      "\n",
      "Writing results 0h 0m 0s 217ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 387ms\n",
      "mergeclusters foldseek_DB foldseek_cluster_0.85/foldseek_cluster_0.85 tmp/9951140192641727792/clu_redundancy tmp/9951140192641727792/clu_step0 tmp/9951140192641727792/clu_step1 tmp/9951140192641727792/clu_step2 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 315.09K 0s 164ms    \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 300.06K 0s 499ms    \n",
      "Clustering step 3\n",
      "[=================================================================] 100.00% 299.88K 0s 782ms    \n",
      "Clustering step 4\n",
      "[=================================================================] 100.00% 299.88K 1s 151ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 542.38K 1s 435ms     \n",
      "Time for merging to foldseek_cluster_0.85: 0h 0m 0s 397ms\n",
      "Time for processing: 0h 0m 2s 301ms\n"
     ]
    }
   ],
   "source": [
    "mkdir foldseek_cluster_0.85\n",
    "foldseek cluster foldseek_DB foldseek_cluster_0.85/foldseek_cluster_0.85 tmp --min-seq-id 0.85"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "53d7ee45-4cb8-4f04-88e8-52f3d429451c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster foldseek_DB foldseek_cluster_0.90/foldseek_cluster_0.90 tmp --min-seq-id 0.90 \n",
      "\n",
      "MMseqs Version:                     \t8.ef4e960\n",
      "Substitution matrix                 \taa:3di.out,nucl:3di.out\n",
      "Seed substitution matrix            \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t1000\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t0\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t1\n",
      "Minimum diagonal score              \t30\n",
      "Selected taxa                       \t\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "TMscore threshold                   \t0\n",
      "LDDT threshold                      \t0\n",
      "Sort by structure bit score         \t0\n",
      "Alignment type                      \t2\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "E-value threshold                   \t0.01\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Gap open cost                       \taa:10,nucl:10\n",
      "Gap extension cost                  \taa:1,nucl:1\n",
      "TMalign hit order                   \t0\n",
      "TMalign fast                        \t1\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t300\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "kmermatcher foldseek_DB_ss tmp/17669332863111538092/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.9 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher foldseek_DB_ss tmp/17669332863111538092/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.9 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 542378 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A F) (C V) (D B) (E Z) (G H) (I M T) (K W) (L J) (N R S) (P) (Q) (Y) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 542.38K 3s 823ms    \n",
      "Sort kmer 0h 0m 9s 575ms\n",
      "Sort by rep. sequence 0h 0m 3s 424ms\n",
      "Time for fill: 0h 0m 1s 818ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 22s 363ms\n",
      "structurerescorediagonal foldseek_DB foldseek_DB tmp/17669332863111538092/pref tmp/17669332863111538092/pref_rescore1 --tmscore-threshold 0 --lddt-threshold 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 542.38K 39s 329ms    \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 291ms\n",
      "Time for processing: 0h 0m 39s 947ms\n",
      "clust foldseek_DB tmp/17669332863111538092/pref_rescore1 tmp/17669332863111538092/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 542.38K 0s 454ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 588806 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 542.38K 0s 473ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 542.38K 0s 116ms    \n",
      "\n",
      "Time for read in: 0h 0m 1s 160ms\n",
      "Total time: 0h 0m 1s 308ms\n",
      "\n",
      "Size of the sequence database: 542378\n",
      "Size of the alignment database: 542378\n",
      "Number of clusters: 343751\n",
      "\n",
      "Writing results 0h 0m 0s 199ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 676ms\n",
      "createsubdb tmp/17669332863111538092/order_redundancy tmp/17669332863111538092/pref tmp/17669332863111538092/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 307ms\n",
      "filterdb tmp/17669332863111538092/pref_filter1 tmp/17669332863111538092/pref_filter2 --filter-file tmp/17669332863111538092/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 343.75K 1s 394ms     \n",
      "Time for merging to pref_filter2: 0h 0m 0s 239ms\n",
      "Time for processing: 0h 0m 1s 872ms\n",
      "structurealign foldseek_DB foldseek_DB tmp/17669332863111538092/pref_filter2 tmp/17669332863111538092/aln.linclust --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 343.75K 10m 7s 119ms    \n",
      "Time for merging to aln.linclust: 0h 0m 0s 192ms\n",
      "Time for processing: 0h 10m 7s 726ms\n",
      "createsubdb tmp/17669332863111538092/order_redundancy foldseek_DB tmp/17669332863111538092/pre_clustered_seqs -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pre_clustered_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 119ms\n",
      "clust tmp/17669332863111538092/pre_clustered_seqs tmp/17669332863111538092/aln.linclust tmp/17669332863111538092/clust.linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 343.75K 0s 286ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 10309 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 343.75K 0s 320ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 343.75K 0s 15ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 685ms\n",
      "Total time: 0h 0m 0s 741ms\n",
      "\n",
      "Size of the sequence database: 343751\n",
      "Size of the alignment database: 343751\n",
      "Number of clusters: 334983\n",
      "\n",
      "Writing results 0h 0m 0s 123ms\n",
      "Time for merging to clust.linclust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 59ms\n",
      "mergeclusters foldseek_DB tmp/17669332863111538092/clu_redundancy tmp/17669332863111538092/pre_clust tmp/17669332863111538092/clust.linclust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 343.75K 0s 123ms    \n",
      "Clustering step 2\n",
      "[==========================================================>      ] 89.50% 299.80K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 334.98K 1s 220ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 334.98K 1s 328ms    \n",
      "Index statistics\n",
      "Entries:          14690476\n",
      "DB size:          572 MB\n",
      "Avg k-mer size:   0.229539\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t5377\n",
      "    SVSHYE\t3296\n",
      "    VVHYEY\t2929\n",
      "    CVSHEY\t2634\n",
      "    IWWWDP\t2484\n",
      "    YEEQVS\t2391\n",
      "    QEEEDP\t2147\n",
      "    AEEEDP\t2070\n",
      "    EEEEDP\t2035\n",
      "    EEEAVV\t1926\n",
      "Time for index table init: 0h 0m 3s 334ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 334983\n",
      "Target db start 1 to 334983\n",
      "[=================================================================] 100.00% 334.98K 48s 499ms    \n",
      "\n",
      "2.810802 k-mers per position\n",
      "31949 DB matches per sequence\n",
      "47 overflows\n",
      "70 sequences passed prefiltering per query sequence\n",
      "100 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 170ms\n",
      "Time for processing: 0h 0m 53s 360ms\n",
      "structurealign tmp/17669332863111538092/input_step_redundancy tmp/17669332863111538092/input_step_redundancy tmp/17669332863111538092/pref_step0 tmp/17669332863111538092/aln_step0 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 334.98K 23m 25s 520ms    \n",
      "Time for merging to aln_step0: 0h 0m 0s 125ms\n",
      "Time for processing: 0h 23m 26s 735ms\n",
      "clust tmp/17669332863111538092/input_step_redundancy tmp/17669332863111538092/aln_step0 tmp/17669332863111538092/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 334.98K 0s 279ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3392 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 334.98K 0s 384ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 334.98K 0s 13ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 724ms\n",
      "Total time: 0h 0m 0s 764ms\n",
      "\n",
      "Size of the sequence database: 334983\n",
      "Size of the alignment database: 334983\n",
      "Number of clusters: 326558\n",
      "\n",
      "Writing results 0h 0m 0s 98ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 932ms\n",
      "createsubdb tmp/17669332863111538092/clu_step0 tmp/17669332863111538092/input_step_redundancy_ss tmp/17669332863111538092/input_step1_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 96ms\n",
      "createsubdb tmp/17669332863111538092/clu_step0 tmp/17669332863111538092/input_step_redundancy_ca tmp/17669332863111538092/input_step1_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 111ms\n",
      "createsubdb tmp/17669332863111538092/clu_step0 tmp/17669332863111538092/input_step_redundancy tmp/17669332863111538092/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 94ms\n",
      "prefilter tmp/17669332863111538092/input_step1_ss tmp/17669332863111538092/input_step1_ss tmp/17669332863111538092/pref_step1 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 200 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 326558 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 326558 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 326.56K 1s 177ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 326.56K 1s 342ms    \n",
      "Index statistics\n",
      "Entries:          14483156\n",
      "DB size:          571 MB\n",
      "Avg k-mer size:   0.226299\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t5167\n",
      "    SVSHYE\t3247\n",
      "    VVHYEY\t2905\n",
      "    CVSHEY\t2618\n",
      "    IWWWDP\t2449\n",
      "    YEEQVS\t2373\n",
      "    QEEEDP\t2131\n",
      "    EEEEDP\t2008\n",
      "    EEEAVV\t1888\n",
      "    SVSHYY\t1755\n",
      "Time for index table init: 0h 0m 3s 162ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 326558\n",
      "Target db start 1 to 326558\n",
      "[=================================================================] 100.00% 326.56K 1m 0s 107ms   \n",
      "\n",
      "1.065154 k-mers per position\n",
      "15285 DB matches per sequence\n",
      "0 overflows\n",
      "121 sequences passed prefiltering per query sequence\n",
      "200 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 173ms\n",
      "Time for processing: 0h 1m 4s 810ms\n",
      "structurealign tmp/17669332863111538092/input_step1 tmp/17669332863111538092/input_step1 tmp/17669332863111538092/pref_step1 tmp/17669332863111538092/aln_step1 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 326.56K 37m 21s 273ms    \n",
      "Time for merging to aln_step1: 0h 0m 0s 126ms\n",
      "Time for processing: 0h 37m 23s 33ms\n",
      "clust tmp/17669332863111538092/input_step1 tmp/17669332863111538092/aln_step1 tmp/17669332863111538092/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 326.56K 0s 293ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 5 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 326.56K 0s 295ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 326.56K 0s 12ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 660ms\n",
      "Total time: 0h 0m 0s 728ms\n",
      "\n",
      "Size of the sequence database: 326558\n",
      "Size of the alignment database: 326558\n",
      "Number of clusters: 326455\n",
      "\n",
      "Writing results 0h 0m 0s 117ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 941ms\n",
      "createsubdb tmp/17669332863111538092/clu_step1 tmp/17669332863111538092/input_step1_ss tmp/17669332863111538092/input_step2_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 109ms\n",
      "createsubdb tmp/17669332863111538092/clu_step1 tmp/17669332863111538092/input_step1_ca tmp/17669332863111538092/input_step2_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 121ms\n",
      "createsubdb tmp/17669332863111538092/clu_step1 tmp/17669332863111538092/input_step1 tmp/17669332863111538092/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 96ms\n",
      "prefilter tmp/17669332863111538092/input_step2_ss tmp/17669332863111538092/input_step2_ss tmp/17669332863111538092/pref_step2 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 326455 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 326455 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 326.45K 1s 128ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 326.45K 1s 311ms    \n",
      "Index statistics\n",
      "Entries:          14481960\n",
      "DB size:          571 MB\n",
      "Avg k-mer size:   0.226281\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t5167\n",
      "    SVSHYE\t3247\n",
      "    VVHYEY\t2905\n",
      "    CVSHEY\t2618\n",
      "    IWWWDP\t2449\n",
      "    YEEQVS\t2373\n",
      "    QEEEDP\t2131\n",
      "    EEEEDP\t2008\n",
      "    EEEAVV\t1888\n",
      "    SVSHYY\t1755\n",
      "Time for index table init: 0h 0m 3s 232ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 326455\n",
      "Target db start 1 to 326455\n",
      "[=================================================================] 100.00% 326.45K 1m 9s 556ms    \n",
      "\n",
      "1.065287 k-mers per position\n",
      "15287 DB matches per sequence\n",
      "0 overflows\n",
      "388 sequences passed prefiltering per query sequence\n",
      "253 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 156ms\n",
      "Time for processing: 0h 1m 14s 273ms\n",
      "structurealign tmp/17669332863111538092/input_step2 tmp/17669332863111538092/input_step2 tmp/17669332863111538092/pref_step2 tmp/17669332863111538092/aln_step2 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 326.45K 1h 12m 18s 282ms \n",
      "Time for merging to aln_step2: 0h 0m 0s 120ms\n",
      "Time for processing: 1h 12m 19s 767ms\n",
      "clust tmp/17669332863111538092/input_step2 tmp/17669332863111538092/aln_step2 tmp/17669332863111538092/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 326.45K 0s 264ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 326.45K 0s 296ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 326.45K 0s 98ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 710ms\n",
      "Total time: 0h 0m 0s 771ms\n",
      "\n",
      "Size of the sequence database: 326455\n",
      "Size of the alignment database: 326455\n",
      "Number of clusters: 326455\n",
      "\n",
      "Writing results 0h 0m 0s 108ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 961ms\n",
      "mergeclusters foldseek_DB foldseek_cluster_0.90/foldseek_cluster_0.90 tmp/17669332863111538092/clu_redundancy tmp/17669332863111538092/clu_step0 tmp/17669332863111538092/clu_step1 tmp/17669332863111538092/clu_step2 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 334.98K 0s 162ms    \n",
      "Clustering step 2\n",
      "[==========================================>                      ] 65.13% 212.69K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[=================================================================] 100.00% 326.45K 1s 126ms    \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 542.38K 1s 550ms     \n",
      "Time for merging to foldseek_cluster_0.90: 0h 0m 0s 340ms\n",
      "Time for processing: 0h 0m 2s 60ms\n"
     ]
    }
   ],
   "source": [
    "mkdir foldseek_cluster_0.90\n",
    "foldseek cluster foldseek_DB foldseek_cluster_0.90/foldseek_cluster_0.90 tmp --min-seq-id 0.90"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "cdc3073d-1017-428b-af31-957cb1b32e77",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster foldseek_DB foldseek_cluster_0.95/foldseek_cluster_0.95 tmp --min-seq-id 0.95 \n",
      "\n",
      "MMseqs Version:                     \t8.ef4e960\n",
      "Substitution matrix                 \taa:3di.out,nucl:3di.out\n",
      "Seed substitution matrix            \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t1000\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t0\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t1\n",
      "Minimum diagonal score              \t30\n",
      "Selected taxa                       \t\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "TMscore threshold                   \t0\n",
      "LDDT threshold                      \t0\n",
      "Sort by structure bit score         \t0\n",
      "Alignment type                      \t2\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "E-value threshold                   \t0.01\n",
      "Seq. id. threshold                  \t0.95\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Gap open cost                       \taa:10,nucl:10\n",
      "Gap extension cost                  \taa:1,nucl:1\n",
      "TMalign hit order                   \t0\n",
      "TMalign fast                        \t1\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t300\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "kmermatcher foldseek_DB_ss tmp/9031565500153207471/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.95 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher foldseek_DB_ss tmp/9031565500153207471/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.95 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 542378 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A F) (C V) (D B) (E Z) (G H) (I M T) (K W) (L J) (N R S) (P) (Q) (Y) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 542.38K 3s 140ms    \n",
      "Sort kmer 0h 0m 7s 985ms\n",
      "Sort by rep. sequence 0h 0m 3s 312ms\n",
      "Time for fill: 0h 0m 1s 682ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 19s 458ms\n",
      "structurerescorediagonal foldseek_DB foldseek_DB tmp/9031565500153207471/pref tmp/9031565500153207471/pref_rescore1 --tmscore-threshold 0 --lddt-threshold 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.95 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 542.38K 39s 339ms     \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 271ms\n",
      "Time for processing: 0h 0m 39s 963ms\n",
      "clust foldseek_DB tmp/9031565500153207471/pref_rescore1 tmp/9031565500153207471/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 542.38K 0s 404ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 423088 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 542.38K 0s 419ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 542.38K 0s 106ms    \n",
      "\n",
      "Time for read in: 0h 0m 1s 39ms\n",
      "Total time: 0h 0m 1s 185ms\n",
      "\n",
      "Size of the sequence database: 542378\n",
      "Size of the alignment database: 542378\n",
      "Number of clusters: 378180\n",
      "\n",
      "Writing results 0h 0m 0s 210ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 540ms\n",
      "createsubdb tmp/9031565500153207471/order_redundancy tmp/9031565500153207471/pref tmp/9031565500153207471/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 296ms\n",
      "filterdb tmp/9031565500153207471/pref_filter1 tmp/9031565500153207471/pref_filter2 --filter-file tmp/9031565500153207471/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 378.18K 1s 598ms     \n",
      "Time for merging to pref_filter2: 0h 0m 0s 297ms\n",
      "Time for processing: 0h 0m 2s 322ms\n",
      "structurealign foldseek_DB foldseek_DB tmp/9031565500153207471/pref_filter2 tmp/9031565500153207471/aln.linclust --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.95 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 378.18K 11m 56s 428ms   \n",
      "Time for merging to aln.linclust: 0h 0m 0s 191ms\n",
      "Time for processing: 0h 11m 56s 940ms\n",
      "createsubdb tmp/9031565500153207471/order_redundancy foldseek_DB tmp/9031565500153207471/pre_clustered_seqs -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pre_clustered_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 178ms\n",
      "clust tmp/9031565500153207471/pre_clustered_seqs tmp/9031565500153207471/aln.linclust tmp/9031565500153207471/clust.linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 378.18K 0s 329ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 4657 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 378.18K 0s 342ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 378.18K 0s 119ms    \n",
      "\n",
      "Time for read in: 0h 0m 0s 884ms\n",
      "Total time: 0h 0m 0s 974ms\n",
      "\n",
      "Size of the sequence database: 378180\n",
      "Size of the alignment database: 378180\n",
      "Number of clusters: 373963\n",
      "\n",
      "Writing results 0h 0m 0s 212ms\n",
      "Time for merging to clust.linclust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 311ms\n",
      "mergeclusters foldseek_DB tmp/9031565500153207471/clu_redundancy tmp/9031565500153207471/pre_clust tmp/9031565500153207471/clust.linclust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 378.18K 1s 260ms    \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 373.96K 1s 534ms     \n",
      "Write merged clustering\n",
      "[===================================>                             ] 55.38% 300.39K eta 1s         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[=================================================================] 100.00% 542.38K 1s 810ms    \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 234ms\n",
      "Time for processing: 0h 0m 2s 206ms\n",
      "createsubdb tmp/9031565500153207471/clu_redundancy foldseek_DB_ss tmp/9031565500153207471/input_step_redundancy_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 149ms\n",
      "createsubdb tmp/9031565500153207471/clu_redundancy foldseek_DB_ca tmp/9031565500153207471/input_step_redundancy_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 149ms\n",
      "createsubdb tmp/9031565500153207471/clu_redundancy foldseek_DB tmp/9031565500153207471/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 122ms\n",
      "prefilter tmp/9031565500153207471/input_step_redundancy_ss tmp/9031565500153207471/input_step_redundancy_ss tmp/9031565500153207471/pref_step0 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 100 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 0 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 373963 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 373963 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 373.96K 1s 865ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 373.96K 1s 967ms    \n",
      "Index statistics\n",
      "Entries:          16510543\n",
      "DB size:          582 MB\n",
      "Avg k-mer size:   0.257977\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t6381\n",
      "    SVSHYE\t3795\n",
      "    VVHYEY\t3271\n",
      "    CVSHEY\t2940\n",
      "    IWWWDP\t2734\n",
      "    YEEQVS\t2646\n",
      "    QEEEDP\t2374\n",
      "    EEEEDP\t2304\n",
      "    KKKFPV\t2244\n",
      "    EEEAVV\t2186\n",
      "Time for index table init: 0h 0m 4s 754ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 373963\n",
      "Target db start 1 to 373963\n",
      "[=================================================================] 100.00% 373.96K 1m 0s 146ms   \n",
      "\n",
      "2.832369 k-mers per position\n",
      "36236 DB matches per sequence\n",
      "93 overflows\n",
      "71 sequences passed prefiltering per query sequence\n",
      "100 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 185ms\n",
      "Time for processing: 0h 1m 6s 363ms\n",
      "structurealign tmp/9031565500153207471/input_step_redundancy tmp/9031565500153207471/input_step_redundancy tmp/9031565500153207471/pref_step0 tmp/9031565500153207471/aln_step0 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.95 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 373.96K 27m 30s 101ms    \n",
      "Time for merging to aln_step0: 0h 0m 0s 305ms\n",
      "Time for processing: 0h 27m 31s 853ms\n",
      "clust tmp/9031565500153207471/input_step_redundancy tmp/9031565500153207471/aln_step0 tmp/9031565500153207471/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 373.96K 0s 367ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2155 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 373.96K 0s 432ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 373.96K 0s 31ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 894ms\n",
      "Total time: 0h 0m 1s 79ms\n",
      "\n",
      "Size of the sequence database: 373963\n",
      "Size of the alignment database: 373963\n",
      "Number of clusters: 365582\n",
      "\n",
      "Writing results 0h 0m 0s 206ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 423ms\n",
      "createsubdb tmp/9031565500153207471/clu_step0 tmp/9031565500153207471/input_step_redundancy_ss tmp/9031565500153207471/input_step1_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 160ms\n",
      "createsubdb tmp/9031565500153207471/clu_step0 tmp/9031565500153207471/input_step_redundancy_ca tmp/9031565500153207471/input_step1_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 152ms\n",
      "createsubdb tmp/9031565500153207471/clu_step0 tmp/9031565500153207471/input_step_redundancy tmp/9031565500153207471/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 152ms\n",
      "prefilter tmp/9031565500153207471/input_step1_ss tmp/9031565500153207471/input_step1_ss tmp/9031565500153207471/pref_step1 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 200 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 365582 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 365582 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 365.58K 1s 436ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 365.58K 1s 830ms    \n",
      "Index statistics\n",
      "Entries:          16293663\n",
      "DB size:          581 MB\n",
      "Avg k-mer size:   0.254588\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t6192\n",
      "    SVSHYE\t3735\n",
      "    VVHYEY\t3236\n",
      "    CVSHEY\t2922\n",
      "    IWWWDP\t2700\n",
      "    YEEQVS\t2621\n",
      "    QEEEDP\t2348\n",
      "    EEEEDP\t2282\n",
      "    EEEAVV\t2151\n",
      "    SVSHYY\t1978\n",
      "Time for index table init: 0h 0m 4s 181ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 365582\n",
      "Target db start 1 to 365582\n",
      "[=================================================================] 100.00% 365.58K 1m 16s 255ms    \n",
      "\n",
      "1.073621 k-mers per position\n",
      "17407 DB matches per sequence\n",
      "1 overflows\n",
      "125 sequences passed prefiltering per query sequence\n",
      "200 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 192ms\n",
      "Time for processing: 0h 1m 22s 417ms\n",
      "structurealign tmp/9031565500153207471/input_step1 tmp/9031565500153207471/input_step1 tmp/9031565500153207471/pref_step1 tmp/9031565500153207471/aln_step1 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.95 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 365.58K 45m 22s 616ms    \n",
      "Time for merging to aln_step1: 0h 0m 0s 227ms\n",
      "Time for processing: 0h 45m 24s 959ms\n",
      "clust tmp/9031565500153207471/input_step1 tmp/9031565500153207471/aln_step1 tmp/9031565500153207471/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 365.58K 0s 311ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 365.58K 0s 372ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 365.58K 0s 13ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 750ms\n",
      "Total time: 0h 0m 0s 821ms\n",
      "\n",
      "Size of the sequence database: 365582\n",
      "Size of the alignment database: 365582\n",
      "Number of clusters: 365441\n",
      "\n",
      "Writing results 0h 0m 0s 115ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 108ms\n",
      "createsubdb tmp/9031565500153207471/clu_step1 tmp/9031565500153207471/input_step1_ss tmp/9031565500153207471/input_step2_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 143ms\n",
      "createsubdb tmp/9031565500153207471/clu_step1 tmp/9031565500153207471/input_step1_ca tmp/9031565500153207471/input_step2_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 106ms\n",
      "createsubdb tmp/9031565500153207471/clu_step1 tmp/9031565500153207471/input_step1 tmp/9031565500153207471/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 123ms\n",
      "prefilter tmp/9031565500153207471/input_step2_ss tmp/9031565500153207471/input_step2_ss tmp/9031565500153207471/pref_step2 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 365441 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 365441 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 365.44K 1s 443ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 365.44K 1s 719ms    \n",
      "Index statistics\n",
      "Entries:          16292525\n",
      "DB size:          581 MB\n",
      "Avg k-mer size:   0.254571\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t6192\n",
      "    SVSHYE\t3735\n",
      "    VVHYEY\t3236\n",
      "    CVSHEY\t2922\n",
      "    IWWWDP\t2700\n",
      "    YEEQVS\t2621\n",
      "    QEEEDP\t2348\n",
      "    EEEEDP\t2282\n",
      "    EEEAVV\t2151\n",
      "    SVSHYY\t1978\n",
      "Time for index table init: 0h 0m 4s 63ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 365441\n",
      "Target db start 1 to 365441\n",
      "[=================================================================] 100.00% 365.44K 1m 24s 926ms    \n",
      "\n",
      "1.073840 k-mers per position\n",
      "17411 DB matches per sequence\n",
      "1 overflows\n",
      "417 sequences passed prefiltering per query sequence\n",
      "303 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 179ms\n",
      "Time for processing: 0h 1m 30s 927ms\n",
      "structurealign tmp/9031565500153207471/input_step2 tmp/9031565500153207471/input_step2 tmp/9031565500153207471/pref_step2 tmp/9031565500153207471/aln_step2 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.95 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 365.44K 3h 35m 48s 74ms     \n",
      "Time for merging to aln_step2: 0h 0m 0s 136ms\n",
      "Time for processing: 3h 35m 50s 45ms\n",
      "clust tmp/9031565500153207471/input_step2 tmp/9031565500153207471/aln_step2 tmp/9031565500153207471/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 365.44K 0s 280ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 365.44K 0s 413ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 365.44K 0s 21ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 762ms\n",
      "Total time: 0h 0m 0s 900ms\n",
      "\n",
      "Size of the sequence database: 365441\n",
      "Size of the alignment database: 365441\n",
      "Number of clusters: 365441\n",
      "\n",
      "Writing results 0h 0m 0s 197ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 188ms\n",
      "mergeclusters foldseek_DB foldseek_cluster_0.95/foldseek_cluster_0.95 tmp/9031565500153207471/clu_redundancy tmp/9031565500153207471/clu_step0 tmp/9031565500153207471/clu_step1 tmp/9031565500153207471/clu_step2 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 373.96K 0s 118ms    \n",
      "Clustering step 2\n",
      "[===========================================>                     ] 66.50% 243.11K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[=================================================================] 100.00% 365.44K 1s 42ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 542.38K 1s 391ms     \n",
      "Time for merging to foldseek_cluster_0.95: 0h 0m 0s 247ms\n",
      "Time for processing: 0h 0m 1s 835ms\n"
     ]
    }
   ],
   "source": [
    "mkdir foldseek_cluster_0.95\n",
    "foldseek cluster foldseek_DB foldseek_cluster_0.95/foldseek_cluster_0.95 tmp --min-seq-id 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "d5d9524f-53a0-40a9-a3ed-b68c9069e878",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster foldseek_DB foldseek_cluster_0.99/foldseek_cluster_0.99 tmp --min-seq-id 0.99 \n",
      "\n",
      "MMseqs Version:                     \t8.ef4e960\n",
      "Substitution matrix                 \taa:3di.out,nucl:3di.out\n",
      "Seed substitution matrix            \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t1000\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t0\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t1\n",
      "Minimum diagonal score              \t30\n",
      "Selected taxa                       \t\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "TMscore threshold                   \t0\n",
      "LDDT threshold                      \t0\n",
      "Sort by structure bit score         \t0\n",
      "Alignment type                      \t2\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "E-value threshold                   \t0.01\n",
      "Seq. id. threshold                  \t0.99\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Gap open cost                       \taa:10,nucl:10\n",
      "Gap extension cost                  \taa:1,nucl:1\n",
      "TMalign hit order                   \t0\n",
      "TMalign fast                        \t1\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t300\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "kmermatcher foldseek_DB_ss tmp/13078304524218135859/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.99 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher foldseek_DB_ss tmp/13078304524218135859/pref --sub-mat 'aa:3di.out,nucl:3di.out' --alph-size aa:21,nucl:5 --min-seq-id 0.99 --kmer-per-seq 300 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 542378 type: Aminoacid\n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 542.38K 3s 415ms    \n",
      "Sort kmer 0h 0m 7s 893ms\n",
      "Sort by rep. sequence 0h 0m 3s 154ms\n",
      "Time for fill: 0h 0m 1s 245ms\n",
      "Time for merging to pref: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 18s 588ms\n",
      "structurerescorediagonal foldseek_DB foldseek_DB tmp/13078304524218135859/pref tmp/13078304524218135859/pref_rescore1 --tmscore-threshold 0 --lddt-threshold 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.99 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 542.38K 38s 580ms    \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 339ms\n",
      "Time for processing: 0h 0m 39s 367ms\n",
      "clust foldseek_DB tmp/13078304524218135859/pref_rescore1 tmp/13078304524218135859/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 542.38K 0s 401ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 283896 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 542.38K 0s 474ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 542.38K 0s 33ms     \n",
      "\n",
      "Time for read in: 0h 0m 1s 67ms\n",
      "Total time: 0h 0m 1s 185ms\n",
      "\n",
      "Size of the sequence database: 542378\n",
      "Size of the alignment database: 542378\n",
      "Number of clusters: 426432\n",
      "\n",
      "Writing results 0h 0m 0s 144ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 508ms\n",
      "createsubdb tmp/13078304524218135859/order_redundancy tmp/13078304524218135859/pref tmp/13078304524218135859/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 300ms\n",
      "filterdb tmp/13078304524218135859/pref_filter1 tmp/13078304524218135859/pref_filter2 --filter-file tmp/13078304524218135859/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 426.43K 1s 711ms     \n",
      "Time for merging to pref_filter2: 0h 0m 0s 278ms\n",
      "Time for processing: 0h 0m 2s 401ms\n",
      "structurealign foldseek_DB foldseek_DB tmp/13078304524218135859/pref_filter2 tmp/13078304524218135859/aln.linclust --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.99 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 426.43K 13m 21s 726ms    \n",
      "Time for merging to aln.linclust: 0h 0m 0s 290ms\n",
      "Time for processing: 0h 13m 22s 333ms\n",
      "createsubdb tmp/13078304524218135859/order_redundancy foldseek_DB tmp/13078304524218135859/pre_clustered_seqs -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pre_clustered_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 140ms\n",
      "clust tmp/13078304524218135859/pre_clustered_seqs tmp/13078304524218135859/aln.linclust tmp/13078304524218135859/clust.linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 426.43K 0s 349ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 532 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 426.43K 0s 324ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 426.43K 0s 125ms    \n",
      "\n",
      "Time for read in: 0h 0m 0s 874ms\n",
      "Total time: 0h 0m 0s 923ms\n",
      "\n",
      "Size of the sequence database: 426432\n",
      "Size of the alignment database: 426432\n",
      "Number of clusters: 425912\n",
      "\n",
      "Writing results 0h 0m 0s 140ms\n",
      "Time for merging to clust.linclust: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 252ms\n",
      "mergeclusters foldseek_DB tmp/13078304524218135859/clu_redundancy tmp/13078304524218135859/pre_clust tmp/13078304524218135859/clust.linclust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 426.43K 0s 129ms    \n",
      "Clustering step 2\n",
      "[===============================================>                 ] 73.29% 312.15K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[=================================================================] 100.00% 542.38K 0s 848ms    \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 245ms\n",
      "Time for processing: 0h 0m 1s 241ms\n",
      "createsubdb tmp/13078304524218135859/clu_redundancy foldseek_DB_ss tmp/13078304524218135859/input_step_redundancy_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 185ms\n",
      "createsubdb tmp/13078304524218135859/clu_redundancy foldseek_DB_ca tmp/13078304524218135859/input_step_redundancy_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 191ms\n",
      "createsubdb tmp/13078304524218135859/clu_redundancy foldseek_DB tmp/13078304524218135859/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 188ms\n",
      "prefilter tmp/13078304524218135859/input_step_redundancy_ss tmp/13078304524218135859/input_step_redundancy_ss tmp/13078304524218135859/pref_step0 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 100 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 0 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 425912 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 425912 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 425.91K 2s 139ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 425.91K 2s 271ms    \n",
      "Index statistics\n",
      "Entries:          18974136\n",
      "DB size:          596 MB\n",
      "Avg k-mer size:   0.296471\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t7639\n",
      "    SVSHYE\t4459\n",
      "    VVHYEY\t3744\n",
      "    CVSHEY\t3369\n",
      "    YEEQVS\t3088\n",
      "    KKKFPV\t2700\n",
      "    AEEEDP\t2600\n",
      "    EEEAVV\t2541\n",
      "    LFEEDP\t2309\n",
      "    SVSHYY\t2267\n",
      "Time for index table init: 0h 0m 5s 452ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 425912\n",
      "Target db start 1 to 425912\n",
      "[=================================================================] 100.00% 425.91K 1m 15s 843ms   \n",
      "\n",
      "2.866611 k-mers per position\n",
      "42012 DB matches per sequence\n",
      "173 overflows\n",
      "72 sequences passed prefiltering per query sequence\n",
      "100 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 190ms\n",
      "Time for processing: 0h 1m 22s 793ms\n",
      "structurealign tmp/13078304524218135859/input_step_redundancy tmp/13078304524218135859/input_step_redundancy tmp/13078304524218135859/pref_step0 tmp/13078304524218135859/aln_step0 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.99 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 425.91K 31m 2s 705ms     \n",
      "Time for merging to aln_step0: 0h 0m 0s 204ms\n",
      "Time for processing: 0h 31m 3s 225ms\n",
      "clust tmp/13078304524218135859/input_step_redundancy tmp/13078304524218135859/aln_step0 tmp/13078304524218135859/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 425.91K 0s 461ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 903 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 425.91K 0s 527ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 425.91K 0s 24ms     \n",
      "\n",
      "Time for read in: 0h 0m 1s 137ms\n",
      "Total time: 0h 0m 1s 330ms\n",
      "\n",
      "Size of the sequence database: 425912\n",
      "Size of the alignment database: 425912\n",
      "Number of clusters: 419638\n",
      "\n",
      "Writing results 0h 0m 0s 209ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 747ms\n",
      "createsubdb tmp/13078304524218135859/clu_step0 tmp/13078304524218135859/input_step_redundancy_ss tmp/13078304524218135859/input_step1_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 212ms\n",
      "createsubdb tmp/13078304524218135859/clu_step0 tmp/13078304524218135859/input_step_redundancy_ca tmp/13078304524218135859/input_step1_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 204ms\n",
      "createsubdb tmp/13078304524218135859/clu_step0 tmp/13078304524218135859/input_step_redundancy tmp/13078304524218135859/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 194ms\n",
      "prefilter tmp/13078304524218135859/input_step1_ss tmp/13078304524218135859/input_step1_ss tmp/13078304524218135859/pref_step1 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 200 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 419638 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 419638 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 419.64K 1s 692ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 419.64K 1s 952ms    \n",
      "Index statistics\n",
      "Entries:          18793751\n",
      "DB size:          595 MB\n",
      "Avg k-mer size:   0.293652\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t7518\n",
      "    SVSHYE\t4402\n",
      "    VVHYEY\t3721\n",
      "    CVSHEY\t3347\n",
      "    YEEQVS\t3064\n",
      "    QEEEDP\t2649\n",
      "    EEEEDP\t2615\n",
      "    EEEAVV\t2509\n",
      "    LFEEDP\t2278\n",
      "    SVSHYY\t2250\n",
      "Time for index table init: 0h 0m 4s 647ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 419638\n",
      "Target db start 1 to 419638\n",
      "[=================================================================] 100.00% 419.64K 1m 41s 720ms    \n",
      "\n",
      "1.088184 k-mers per position\n",
      "20323 DB matches per sequence\n",
      "2 overflows\n",
      "129 sequences passed prefiltering per query sequence\n",
      "200 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 204ms\n",
      "Time for processing: 0h 1m 48s 426ms\n",
      "structurealign tmp/13078304524218135859/input_step1 tmp/13078304524218135859/input_step1 tmp/13078304524218135859/pref_step1 tmp/13078304524218135859/aln_step1 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.99 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 419.64K 54m 27s 469ms     \n",
      "Time for merging to aln_step1: 0h 0m 0s 269ms\n",
      "Time for processing: 0h 54m 30s 567ms\n",
      "clust tmp/13078304524218135859/input_step1 tmp/13078304524218135859/aln_step1 tmp/13078304524218135859/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 419.64K 0s 318ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 6 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 419.64K 0s 279ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 419.64K 0s 172ms    \n",
      "\n",
      "Time for read in: 0h 0m 0s 916ms\n",
      "Total time: 0h 0m 0s 990ms\n",
      "\n",
      "Size of the sequence database: 419638\n",
      "Size of the alignment database: 419638\n",
      "Number of clusters: 419522\n",
      "\n",
      "Writing results 0h 0m 0s 240ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 362ms\n",
      "createsubdb tmp/13078304524218135859/clu_step1 tmp/13078304524218135859/input_step1_ss tmp/13078304524218135859/input_step2_ss -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ss: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 164ms\n",
      "createsubdb tmp/13078304524218135859/clu_step1 tmp/13078304524218135859/input_step1_ca tmp/13078304524218135859/input_step2_ca -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2_ca: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 272ms\n",
      "createsubdb tmp/13078304524218135859/clu_step1 tmp/13078304524218135859/input_step1 tmp/13078304524218135859/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 186ms\n",
      "prefilter tmp/13078304524218135859/input_step2_ss tmp/13078304524218135859/input_step2_ss tmp/13078304524218135859/pref_step2 --sub-mat 'aa:3di.out,nucl:3di.out' --seed-sub-mat 'aa:3di.out,nucl:3di.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 1000 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 0.15 --diag-score 1 --exact-kmer-matching 0 --mask 0 --mask-prob 0.9 --mask-lower-case 1 --min-ungapped-score 60 --add-self-matches 1 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 419522 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 419522 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 419.52K 1s 665ms    \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 419.52K 1s 591ms    \n",
      "Index statistics\n",
      "Entries:          18791504\n",
      "DB size:          595 MB\n",
      "Avg k-mer size:   0.293617\n",
      "Top 10 k-mers\n",
      "    EEEEPL\t7518\n",
      "    SVSHYE\t4402\n",
      "    VVHYEY\t3721\n",
      "    CVSHEY\t3347\n",
      "    YEEQVS\t3064\n",
      "    QEEEDP\t2649\n",
      "    EEEEDP\t2615\n",
      "    EEEAVV\t2509\n",
      "    LFEEDP\t2278\n",
      "    SVSHYY\t2250\n",
      "Time for index table init: 0h 0m 4s 94ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 419522\n",
      "Target db start 1 to 419522\n",
      "[=================================================================] 100.00% 419.52K 1m 49s 217ms    \n",
      "\n",
      "1.088195 k-mers per position\n",
      "20321 DB matches per sequence\n",
      "2 overflows\n",
      "453 sequences passed prefiltering per query sequence\n",
      "375 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 182ms\n",
      "Time for processing: 0h 1m 55s 3ms\n",
      "structurealign tmp/13078304524218135859/input_step2 tmp/13078304524218135859/input_step2 tmp/13078304524218135859/pref_step2 tmp/13078304524218135859/aln_step2 --tmscore-threshold 0 --lddt-threshold 0 --sort-by-structure-bits 0 --alignment-type 2 --sub-mat 'aa:3di.out,nucl:3di.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.01 --min-seq-id 0.99 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:10,nucl:10 --gap-extend aa:1,nucl:1 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 419.52K 2h 23m 12s 804ms    \n",
      "Time for merging to aln_step2: 0h 0m 0s 535ms\n",
      "Time for processing: 2h 23m 15s 965ms\n",
      "clust tmp/13078304524218135859/input_step2 tmp/13078304524218135859/aln_step2 tmp/13078304524218135859/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 419.52K 0s 311ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 419.52K 0s 402ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 419.52K 0s 36ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 919ms\n",
      "Total time: 0h 0m 0s 994ms\n",
      "\n",
      "Size of the sequence database: 419522\n",
      "Size of the alignment database: 419522\n",
      "Number of clusters: 419519\n",
      "\n",
      "Writing results 0h 0m 0s 238ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 503ms\n",
      "mergeclusters foldseek_DB foldseek_cluster_0.99/foldseek_cluster_0.99 tmp/13078304524218135859/clu_redundancy tmp/13078304524218135859/clu_step0 tmp/13078304524218135859/clu_step1 tmp/13078304524218135859/clu_step2 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 425.91K 0s 169ms    \n",
      "Clustering step 2\n",
      "[==================================================>              ] 78.09% 327.70K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[=================================================================] 100.00% 419.52K 1s 159ms    \n",
      "Clustering step 4\n",
      "[=================================================================] 100.00% 419.52K 1s 766ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 542.38K 2s 48ms       \n",
      "Time for merging to foldseek_cluster_0.99: 0h 0m 0s 308ms\n",
      "Time for processing: 0h 0m 3s 17ms\n"
     ]
    }
   ],
   "source": [
    "mkdir foldseek_cluster_0.99\n",
    "foldseek cluster foldseek_DB foldseek_cluster_0.99/foldseek_cluster_0.99 tmp --min-seq-id 0.99"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3659456b-591f-4f12-9fbc-4996a50d877c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# _rep_seq.fasta --> representative sequences. How many clusters it made.\n",
    "# _cluster.tsv --> representitive - member"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af5a3865-043f-453c-a72e-20040d8d3742",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6690117-2a1e-4cd1-9d84-6e97ac025fa1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e869929e-ca37-4279-adaf-6d68f12dd76b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createtsv foldseek_DB foldseek_DB foldseek_cluster_0.85/foldseek_cluster_0.85 foldseek_cluster_0.85/foldseek_cluster_0.85.tsv \n",
      "\n",
      "MMseqs Version:                 \t8.ef4e960\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t8\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to foldseek_cluster_0.85.tsv: 0h 0m 0s 189ms\n",
      "Time for processing: 0h 0m 0s 530ms\n"
     ]
    }
   ],
   "source": [
    "foldseek createtsv foldseek_DB foldseek_DB foldseek_cluster_0.85/foldseek_cluster_0.85 foldseek_cluster_0.85/foldseek_cluster_0.85.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "1658d3b2-eb7f-4360-b351-7f32c20ae329",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createtsv foldseek_DB foldseek_DB foldseek_cluster_0.90/foldseek_cluster_0.90 foldseek_cluster_0.90/foldseek_cluster_0.90.tsv \n",
      "\n",
      "MMseqs Version:                 \t8.ef4e960\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t8\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to foldseek_cluster_0.90.tsv: 0h 0m 0s 148ms\n",
      "Time for processing: 0h 0m 0s 382ms\n"
     ]
    }
   ],
   "source": [
    "foldseek createtsv foldseek_DB foldseek_DB foldseek_cluster_0.90/foldseek_cluster_0.90 foldseek_cluster_0.90/foldseek_cluster_0.90.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "4ae459bf-423b-4ef6-b10e-e40c8af9f77f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createtsv foldseek_DB foldseek_DB foldseek_cluster_0.95/foldseek_cluster_0.95 foldseek_cluster_0.95/foldseek_cluster_0.95.tsv \n",
      "\n",
      "MMseqs Version:                 \t8.ef4e960\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t8\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to foldseek_cluster_0.95.tsv: 0h 0m 0s 158ms\n",
      "Time for processing: 0h 0m 0s 413ms\n"
     ]
    }
   ],
   "source": [
    "foldseek createtsv foldseek_DB foldseek_DB foldseek_cluster_0.95/foldseek_cluster_0.95 foldseek_cluster_0.95/foldseek_cluster_0.95.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "51e4d1b7-15fe-4dc4-84d1-73b7dcc14ccb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createtsv foldseek_DB foldseek_DB foldseek_cluster_0.99/foldseek_cluster_0.99 foldseek_cluster_0.99/foldseek_cluster_0.99.tsv \n",
      "\n",
      "MMseqs Version:                 \t8.ef4e960\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t8\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to foldseek_cluster_0.99.tsv: 0h 0m 0s 209ms\n",
      "Time for processing: 0h 0m 0s 454ms\n"
     ]
    }
   ],
   "source": [
    "foldseek createtsv foldseek_DB foldseek_DB foldseek_cluster_0.99/foldseek_cluster_0.99 foldseek_cluster_0.99/foldseek_cluster_0.99.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5843903-652b-4e61-b7ab-65a8cc8ab4c8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "2cf32274-b9dc-4562-a05b-a9391c450bab",
   "metadata": {},
   "source": [
    "# Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ab76ea96-177a-4632-ac73-151c01c2f3d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir foldseek_search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f92922bf-3892-4d36-9eaf-3b78e198a46e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEADER    STRUCTURAL GENOMICS, UNKNOWN FUNCTION   16-DEC-13   2MIO              \n",
      "TITLE     SOLUTION NMR STRUCTURE OF SH3 DOMAIN 1 OF RHO GTPASE-ACTIVATING       \n",
      "TITLE    2 PROTEIN 10 FROM HOMO SAPIENS, NORTHEAST STRUCTURAL GENOMICS          \n",
      "TITLE    3 CONSORTIUM (NESG) TARGET HR9129A                                     \n",
      "COMPND    MOL_ID: 1;                                                            \n",
      "COMPND   2 MOLECULE: RHO GTPASE-ACTIVATING PROTEIN 10;                          \n",
      "COMPND   3 CHAIN: A;                                                            \n",
      "COMPND   4 FRAGMENT: SH3 DOMAIN 1 RESIDUES 728-786;                             \n",
      "COMPND   5 SYNONYM: GTPASE REGULATOR ASSOCIATED WITH FOCAL ADHESION KINASE 2,   \n",
      "COMPND   6 GRAF-RELATED PROTEIN 2, RHO-TYPE GTPASE-ACTIVATING PROTEIN 10;       \n"
     ]
    }
   ],
   "source": [
    "head 2mio.pdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5a9fe89d-c0db-4ba9-8ea1-5b87ff32f21d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create directory tmpFolder\n",
      "easy-search 2mio.pdb foldseek_search/ 2mio tmpFolder \n",
      "\n",
      "MMseqs Version:              \t8.ef4e960\n",
      "Seq. id. threshold           \t0\n",
      "Coverage threshold           \t0\n",
      "Coverage mode                \t0\n",
      "Max reject                   \t2147483647\n",
      "Max accept                   \t2147483647\n",
      "Add backtrace                \tfalse\n",
      "TMscore threshold            \t0\n",
      "TMalign hit order            \t0\n",
      "TMalign fast                 \t1\n",
      "Preload mode                 \t0\n",
      "Threads                      \t8\n",
      "Verbosity                    \t3\n",
      "LDDT threshold               \t0\n",
      "Sort by structure bit score  \t1\n",
      "Alignment type               \t2\n",
      "Substitution matrix          \taa:3di.out,nucl:3di.out\n",
      "Alignment mode               \t3\n",
      "Alignment mode               \t0\n",
      "E-value threshold            \t10\n",
      "Min alignment length         \t0\n",
      "Seq. id. mode                \t0\n",
      "Alternative alignments       \t0\n",
      "Max sequence length          \t65535\n",
      "Compositional bias           \t1\n",
      "Compositional bias           \t1\n",
      "Gap open cost                \taa:10,nucl:10\n",
      "Gap extension cost           \taa:1,nucl:1\n",
      "Compressed                   \t0\n",
      "Seed substitution matrix     \taa:3di.out,nucl:3di.out\n",
      "Sensitivity                  \t9.5\n",
      "k-mer length                 \t6\n",
      "Target search mode           \t0\n",
      "k-score                      \tseq:2147483647,prof:2147483647\n",
      "Max results per query        \t1000\n",
      "Split database               \t0\n",
      "Split mode                   \t2\n",
      "Split memory limit           \t0\n",
      "Diagonal scoring             \ttrue\n",
      "Exact k-mer matching         \t0\n",
      "Mask residues                \t0\n",
      "Mask residues probability    \t0.99995\n",
      "Mask lower case residues     \t1\n",
      "Minimum diagonal score       \t30\n",
      "Selected taxa                \t\n",
      "Spaced k-mers                \t1\n",
      "Spaced k-mer pattern         \t\n",
      "Local temporary path         \t\n",
      "Exhaustive search mode       \tfalse\n",
      "Prefilter mode               \t0\n",
      "Search iterations            \t1\n",
      "Remove temporary files       \ttrue\n",
      "MPI runner                   \t\n",
      "Force restart with latest tmp\tfalse\n",
      "Cluster search               \t0\n",
      "Chain name mode              \t0\n",
      "Write mapping file           \t0\n",
      "Mask b-factor threshold      \t0\n",
      "Coord store mode             \t2\n",
      "Write lookup file            \t1\n",
      "Tar Inclusion Regex          \t.*\n",
      "Tar Exclusion Regex          \t^$\n",
      "File Inclusion Regex         \t.*\n",
      "File Exclusion Regex         \t^$\n",
      "Alignment format             \t0\n",
      "Format alignment output      \tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Database output              \tfalse\n",
      "Greedy best hits             \tfalse\n",
      "\n",
      "createdb 2mio.pdb tmpFolder/6392885911727684435/query --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 8 -v 3 \n",
      "\n",
      "Output file: tmpFolder/6392885911727684435/query\n",
      "[=================================================================] 100.00% 1 eta -\n",
      "Time for merging to query_ss: 0h 0m 0s 9ms\n",
      "Time for merging to query_h: 0h 0m 0s 0ms\n",
      "Time for merging to query_ca: 0h 0m 0s 1ms\n",
      "Time for merging to query: 0h 0m 0s 0ms\n",
      "Ignore 0 out of 20.\n",
      "Too short: 0, incorrect: 0, not proteins: 0.\n",
      "Time for processing: 0h 0m 0s 110ms\n",
      "createdb foldseek_search/ tmpFolder/6392885911727684435/target --chain-name-mode 0 --write-mapping 0 --mask-bfactor-threshold 0 --coord-store-mode 2 --write-lookup 1 --tar-include '.*' --tar-exclude '^$' --file-include '.*' --file-exclude '^$' --threads 8 -v 3 \n",
      "\n",
      "Output file: tmpFolder/6392885911727684435/target\n",
      "Time for merging to target_ss: 0h 0m 0s 1ms\n",
      "Time for merging to target_h: 0h 0m 0s 1ms\n",
      "Time for merging to target_ca: 0h 0m 0s 1ms\n",
      "Time for merging to target: 0h 0m 0s 0ms\n",
      "\u001b[31mNo structures found in given input.\n",
      "\u001b[39mError: target createdb died\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "foldseek easy-search 2mio.pdb foldseek_DB foldseek_search/2mio_aln2 tmpFolder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0769d7a7-47de-41fe-a3a0-53e9a68fa310",
   "metadata": {},
   "outputs": [],
   "source": [
    "head foldseek_search/2mio_aln2\n",
    "# query,target,fident(Fraction of identical matches),alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71947506-b503-422f-8426-07df033cf645",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "215dbc9d-2be8-4b8d-8724-f7e38ba35b7d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
