{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "74237baa-5382-45f4-8619-dfb3833fffc9",
   "metadata": {},
   "source": [
    "# Clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b960b3b9-b40c-4eec-b1ea-134b94342a4c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6-all-vs-all-similarity-queryId_targetId_eValue.tsv  SwissProt.tsv   \u001b[0m\u001b[01;34mmmseqs\u001b[0m\n",
      "IBP.ipynb                                            Untitled.ipynb  \u001b[01;34msnap\u001b[0m\n",
      "SwissProt.fasta                                      \u001b[01;34mfoldseek\u001b[0m        \u001b[01;34mtmp\u001b[0m\n",
      "\u001b[01;31mSwissProt.fasta.gz\u001b[0m                                   \u001b[01;34mminiconda3\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a28b6ba7-ff75-4112-982f-be18a2066004",
   "metadata": {},
   "outputs": [],
   "source": [
    "# mkdir MMseqs2_cluster_0.99\n",
    "# mkdir MMseqs2_cluster_0.95\n",
    "# mkdir MMseqs2_cluster_0.90\n",
    "# mkdir MMseqs2_cluster_0.85\n",
    "# mkdir MMseqs2_cluster_0.80"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "10a8c97d-d5c2-4fbc-a875-b7d067fda1e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb SwissProt.fasta MMseqs2_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[570146] 6s 392ms\n",
      "Time for merging to MMseqs2_DB_h: 0h 0m 0s 797ms\n",
      "Time for merging to MMseqs2_DB: 0h 0m 2s 74ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 11s 255ms\n"
     ]
    }
   ],
   "source": [
    "# Before clustering, convert your FASTA database into the MMseqs2 database (DB) format\n",
    "mmseqs createdb SwissProt.fasta MMseqs2_DB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4504b545-f2a1-4de5-b53c-12f8bca0fa2d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster MMseqs2_DB MMseqs2_cluster tmp --min-seq-id 0.9 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 1\n",
      "linclust MMseqs2_DB tmp/16027832657523067577/clu_redundancy tmp/16027832657523067577/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher MMseqs2_DB tmp/16027832657523067577/linclust/12652008710893151048/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher MMseqs2_DB tmp/16027832657523067577/linclust/12652008710893151048/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 570157 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 570.16K 2s 220ms    \n",
      "Sort kmer 0h 0m 0s 586ms\n",
      "Sort by rep. sequence 0h 0m 0s 306ms\n",
      "Time for fill: 0h 0m 0s 238ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 794ms\n",
      "rescorediagonal MMseqs2_DB MMseqs2_DB tmp/16027832657523067577/linclust/12652008710893151048/pref tmp/16027832657523067577/linclust/12652008710893151048/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 570.16K 0s 391ms    \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 276ms================>] 99.38% 566.60K eta 0s       \n",
      "Time for processing: 0h 0m 1s 20ms\n",
      "clust MMseqs2_DB tmp/16027832657523067577/linclust/12652008710893151048/pref_rescore1 tmp/16027832657523067577/linclust/12652008710893151048/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 570.16K 0s 390ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 359071 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 570.16K 0s 356ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 570.16K 0s 27ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 951ms\n",
      "Total time: 0h 0m 1s 74ms\n",
      "\n",
      "Size of the sequence database: 570157\n",
      "Size of the alignment database: 570157\n",
      "Number of clusters: 351417\n",
      "\n",
      "Writing results 0h 0m 0s 138ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 383ms\n",
      "createsubdb tmp/16027832657523067577/linclust/12652008710893151048/order_redundancy MMseqs2_DB tmp/16027832657523067577/linclust/12652008710893151048/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 81ms\n",
      "createsubdb tmp/16027832657523067577/linclust/12652008710893151048/order_redundancy tmp/16027832657523067577/linclust/12652008710893151048/pref tmp/16027832657523067577/linclust/12652008710893151048/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 296ms\n",
      "filterdb tmp/16027832657523067577/linclust/12652008710893151048/pref_filter1 tmp/16027832657523067577/linclust/12652008710893151048/pref_filter2 --filter-file tmp/16027832657523067577/linclust/12652008710893151048/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 351.42K 0s 287ms    \n",
      "Time for merging to pref_filter2: 0h 0m 0s 180ms\n",
      "Time for processing: 0h 0m 0s 741ms\n",
      "rescorediagonal tmp/16027832657523067577/linclust/12652008710893151048/input_step_redundancy tmp/16027832657523067577/linclust/12652008710893151048/input_step_redundancy tmp/16027832657523067577/linclust/12652008710893151048/pref_filter2 tmp/16027832657523067577/linclust/12652008710893151048/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[==================================>                              ] 53.33% 187.42K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[=================================================================] 100.00% 351.42K 23s 514ms    \n",
      "Time for merging to aln: 0h 0m 0s 196ms\n",
      "595950 alignments calculated\n",
      "362700 sequence pairs passed the thresholds (0.608608 of overall calculated)\n",
      "1.032107 hits per query sequence\n",
      "Time for processing: 0h 0m 23s 873ms\n",
      "clust tmp/16027832657523067577/linclust/12652008710893151048/input_step_redundancy tmp/16027832657523067577/linclust/12652008710893151048/aln tmp/16027832657523067577/linclust/12652008710893151048/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 351.42K 0s 260ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 11283 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 351.42K 0s 267ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 351.42K 0s 95ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 680ms\n",
      "Total time: 0h 0m 0s 740ms\n",
      "\n",
      "Size of the sequence database: 351417\n",
      "Size of the alignment database: 351417\n",
      "Number of clusters: 340981\n",
      "\n",
      "Writing results 0h 0m 0s 198ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 23ms\n",
      "mergeclusters MMseqs2_DB tmp/16027832657523067577/clu_redundancy tmp/16027832657523067577/linclust/12652008710893151048/pre_clust tmp/16027832657523067577/linclust/12652008710893151048/clust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 351.42K 0s 144ms    \n",
      "Clustering step 2\n",
      "[=========================>                                       ] 39.36% 134.21K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[=================================================================] 100.00% 340.98K 6s 516ms    \n",
      "Index table: Masked residues: 2540421\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 340.98K 4s 485ms    \n",
      "Index statistics\n",
      "Entries:          57260585\n",
      "DB size:          815 MB\n",
      "Avg k-mer size:   0.894697\n",
      "Top 10 k-mers\n",
      "    GPGGTL\t1061\n",
      "    PGVFEV\t713\n",
      "    GQSWTV\t533\n",
      "    GYLWMS\t502\n",
      "    HIRVGW\t494\n",
      "    VLWQFW\t478\n",
      "    TPHKWY\t441\n",
      "    RRDQGR\t432\n",
      "    TGFAYT\t427\n",
      "    IMHDKT\t422\n",
      "Time for index table init: 0h 0m 11s 825ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 340981\n",
      "Target db start 1 to 340981\n",
      "[=================================================================] 100.00% 340.98K 21s 798ms    \n",
      "\n",
      "1.866505 k-mers per position\n",
      "1904 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "14 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 170ms\n",
      "Time for processing: 0h 0m 35s 447ms\n",
      "align tmp/16027832657523067577/input_step_redundancy tmp/16027832657523067577/input_step_redundancy tmp/16027832657523067577/pref_step0 tmp/16027832657523067577/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 340981 type: Aminoacid\n",
      "Target database size: 340981 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 340.98K 10m 10s 791ms   \n",
      "Time for merging to aln_step0: 0h 0m 0s 168ms\n",
      "4182069 alignments calculated\n",
      "361431 sequence pairs passed the thresholds (0.086424 of overall calculated)\n",
      "1.059974 hits per query sequence\n",
      "Time for processing: 0h 10m 11s 121ms\n",
      "clust tmp/16027832657523067577/input_step_redundancy tmp/16027832657523067577/aln_step0 tmp/16027832657523067577/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 340.98K 0s 490ms    \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1102 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 340.98K 0s 317ms    \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 340.98K 0s 13ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 887ms\n",
      "Total time: 0h 0m 0s 974ms\n",
      "\n",
      "Size of the sequence database: 340981\n",
      "Size of the alignment database: 340981\n",
      "Number of clusters: 334561\n",
      "\n",
      "Writing results 0h 0m 0s 167ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 231ms\n",
      "mergeclusters MMseqs2_DB MMseqs2_cluster tmp/16027832657523067577/clu_redundancy tmp/16027832657523067577/clu_step0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 340.98K 0s 150ms    \n",
      "Clustering step 2\n",
      "[============================================================>    ] 92.39% 309.11K eta 0s       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Cluster MMseqs2_DB\n",
    "# Better to run the code on the terminal, avoiding \"IOPub message rate exceeded\" warning.\n",
    "mmseqs cluster MMseqs2_DB MMseqs2_cluster_0.99/MMseqs2_cluster_0.99 tmp --min-seq-id 0.99\n",
    "\n",
    "# mmseqs cluster MMseqs2_DB MMseqs2_cluster_0.95/MMseqs2_cluster_0.95 tmp --min-seq-id 0.95\n",
    "# mmseqs cluster MMseqs2_DB MMseqs2_cluster_0.90/MMseqs2_cluster_0.90 tmp --min-seq-id 0.90\n",
    "# mmseqs cluster MMseqs2_DB MMseqs2_cluster_0.85/MMseqs2_cluster_0.85 tmp --min-seq-id 0.85\n",
    "# mmseqs cluster MMseqs2_DB MMseqs2_cluster_0.80/MMseqs2_cluster_0.80 tmp --min-seq-id 0.80"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba9257f1-c396-41a0-b795-bdacfeee392c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert the internal format to a flat tsv file\n",
    "\n",
    "mmseqs createtsv MMseqs2_DB MMseqs2_DB MMseqs2_cluster_0.99/MMseqs2_cluster_0.99 MMseqs2_cluster_0.99/MMseqs2_cluster_0.99.tsv\n",
    "# mmseqs createtsv MMseqs2_DB MMseqs2_DB MMseqs2_cluster_0.95/MMseqs2_cluster_0.95 MMseqs2_cluster_0.95/MMseqs2_cluster_0.95.tsv\n",
    "# mmseqs createtsv MMseqs2_DB MMseqs2_DB MMseqs2_cluster_0.90/MMseqs2_cluster_0.90 MMseqs2_cluster_0.90/MMseqs2_cluster_0.90.tsv\n",
    "# mmseqs createtsv MMseqs2_DB MMseqs2_DB MMseqs2_cluster_0.85/MMseqs2_cluster_0.85 MMseqs2_cluster_0.85/MMseqs2_cluster_0.85.tsv\n",
    "# mmseqs createtsv MMseqs2_DB MMseqs2_DB MMseqs2_cluster_0.80/MMseqs2_cluster_0.80 MMseqs2_cluster_0.80/MMseqs2_cluster_0.80.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d4f6635-0bfa-4595-8f89-b41b0cdff619",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c6ef227d-51b3-43b5-aa36-50e360cdc229",
   "metadata": {},
   "source": [
    "# Searching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0196b3d1-5a2e-4dd9-884d-2133b1bac136",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">sp|A0A0C5B5G6|MOTSC_HUMAN Mitochondrial-derived peptide MOTS-c OS=Homo sapiens OX=9606 GN=MT-RNR1 PE=1 SV=1\n",
      "MRWQEMGYIFYPRKLR\n"
     ]
    }
   ],
   "source": [
    "head QUERY.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7e8e7176-8811-4dcf-903a-eb07001f2099",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">sp|Q6GZX4|001R_FRG3G Putative transcription factor 001R OS=Frog virus 3 (isolate Goorha) OX=654924 GN=FV3-001R PE=4 SV=1\n",
      "MAFSAEDVLKEYDRRRRMEALLLSLYYPNDRKLLDYKEWSPPRVQVECPKAPVEWNNPPS\n",
      "EKGLIVGHFSGIKYKGEKAQASEVDVNKMCCWVSKFKDAMRRYQGIQTCKIPGKVLSDLD\n",
      "AKIKAYNLTVEGVEGFVRYSRVTKQHVAAFLKELRHSKQYENVNLIHYILTDKRVDIQHL\n",
      "EKDLVKDFKALVESAHRMRQGHMINVKYILYQLLKKHGHGPDGPDILTVKTGSKGVLYDD\n",
      "SFRKIYTDLGWKFTPL\n",
      ">sp|Q6GZX3|002L_FRG3G Uncharacterized protein 002L OS=Frog virus 3 (isolate Goorha) OX=654924 GN=FV3-002L PE=4 SV=1\n",
      "MSIIGATRLQNDKSDTYSAGPCYAGGCSAFTPRGTCGKDWDLGEQTCASGFCTSQPLCAR\n",
      "IKKTQVCGLRYSSKGKDPLVSAEWDSRGAPYVRCTYDADLIDTQAQVDQFVSMFGESPSL\n",
      "AERYCMRGVKNTAGELVSRVSSDADPAGGWCRKWYSAHRGPDQDAALGSFCIKNPGAADC\n"
     ]
    }
   ],
   "source": [
    "head SwissProt.fasta #targetDB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b0642c12-8dfa-4810-9500-e817e280fb3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir queryDB\n",
    "mkdir targetDB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "aa4c8460-4f44-419d-b32e-471e5be93420",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb QUERY.fasta queryDB/queryDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "\n",
      "Time for merging to queryDB_h: 0h 0m 0s 11ms\n",
      "Time for merging to queryDB: 0h 0m 0s 1ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 30ms\n"
     ]
    }
   ],
   "source": [
    "# convert your FASTA file into a sequence DB.\n",
    "mmseqs createdb QUERY.fasta queryDB/queryDB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "325ee5a7-45d5-4cbf-989c-cd5da2a72977",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 32\n",
      "-rw-r--r-- 1 jenny352013 jenny352013  18 Oct 18 11:18 queryDB\n",
      "-rw-r--r-- 1 jenny352013 jenny352013   4 Oct 18 11:18 queryDB.dbtype\n",
      "-rw-r--r-- 1 jenny352013 jenny352013   7 Oct 18 11:18 queryDB.index\n",
      "-rw-r--r-- 1 jenny352013 jenny352013  15 Oct 18 11:18 queryDB.lookup\n",
      "-rw-r--r-- 1 jenny352013 jenny352013  14 Oct 18 11:18 queryDB.source\n",
      "-rw-r--r-- 1 jenny352013 jenny352013 109 Oct 18 11:18 queryDB_h\n",
      "-rw-r--r-- 1 jenny352013 jenny352013   4 Oct 18 11:18 queryDB_h.dbtype\n",
      "-rw-r--r-- 1 jenny352013 jenny352013   8 Oct 18 11:18 queryDB_h.index\n"
     ]
    }
   ],
   "source": [
    "ls -l queryDB/\n",
    "# The queryDB and queryDB.index files contain the amino acid sequences, \n",
    "# The queryDB_h and queryDB_h.index file contain the FASTA headers. \n",
    "# The queryDB.lookup file contains a list of tab separated fields that map from the internal identifier to the FASTA identifiers."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a3f1bbeb-aa41-4a2a-a685-24eb785bbfac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb SwissProt.fasta targetDB/targetDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[301789] 1s 6msms\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[570146] 1s 784ms\n",
      "Time for merging to targetDB_h: 0h 0m 0s 229ms\n",
      "Time for merging to targetDB: 0h 0m 0s 456ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 4s 81ms\n"
     ]
    }
   ],
   "source": [
    "mmseqs createdb SwissProt.fasta targetDB/targetDB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "356e521f-e338-4c6e-845e-669dc994f355",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 305916\n",
      "-rw-r--r-- 1 jenny352013 jenny352013 207313693 Oct 18 11:22 targetDB\n",
      "-rw-r--r-- 1 jenny352013 jenny352013         4 Oct 18 11:22 targetDB.dbtype\n",
      "-rw-r--r-- 1 jenny352013 jenny352013  11522221 Oct 18 11:22 targetDB.index\n",
      "-rw-r--r-- 1 jenny352013 jenny352013   9028842 Oct 18 11:22 targetDB.lookup\n",
      "-rw-r--r-- 1 jenny352013 jenny352013        18 Oct 18 11:22 targetDB.source\n",
      "-rw-r--r-- 1 jenny352013 jenny352013  74232078 Oct 18 11:22 targetDB_h\n",
      "-rw-r--r-- 1 jenny352013 jenny352013         4 Oct 18 11:22 targetDB_h.dbtype\n",
      "-rw-r--r-- 1 jenny352013 jenny352013  11126244 Oct 18 11:22 targetDB_h.index\n"
     ]
    }
   ],
   "source": [
    "ls -l targetDB/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a2a14a22-2889-4408-9999-7b79e4f58386",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createindex targetDB/targetDB tmp \n",
      "\n",
      "MMseqs Version:          \t14.7e284\n",
      "Seed substitution matrix \taa:VTML80.out,nucl:nucleotide.out\n",
      "k-mer length             \t0\n",
      "Alphabet size            \taa:21,nucl:5\n",
      "Compositional bias       \t1\n",
      "Compositional bias       \t1\n",
      "Max sequence length      \t65535\n",
      "Max results per query    \t300\n",
      "Mask residues            \t1\n",
      "Mask residues probability\t0.9\n",
      "Mask lower case residues \t0\n",
      "Spaced k-mers            \t1\n",
      "Spaced k-mer pattern     \t\n",
      "Sensitivity              \t7.5\n",
      "k-score                  \tseq:0,prof:0\n",
      "Check compatible         \t0\n",
      "Search type              \t0\n",
      "Split database           \t0\n",
      "Split memory limit       \t0\n",
      "Verbosity                \t3\n",
      "Threads                  \t8\n",
      "Min codons in orf        \t30\n",
      "Max codons in length     \t32734\n",
      "Max orf gaps             \t2147483647\n",
      "Contig start mode        \t2\n",
      "Contig end mode          \t2\n",
      "Orf start mode           \t1\n",
      "Forward frames           \t1,2,3\n",
      "Reverse frames           \t1,2,3\n",
      "Translation table        \t1\n",
      "Translate orf            \t0\n",
      "Use all table starts     \tfalse\n",
      "Offset of numeric ids    \t0\n",
      "Create lookup            \t0\n",
      "Compressed               \t0\n",
      "Add orf stop             \tfalse\n",
      "Overlap between sequences\t0\n",
      "Sequence split mode      \t1\n",
      "Header split mode        \t0\n",
      "Strand selection         \t1\n",
      "Remove temporary files   \tfalse\n",
      "\n",
      "createindex targetDB/targetDB tmp \n",
      "\n",
      "MMseqs Version:          \t14.7e284\n",
      "Seed substitution matrix \taa:VTML80.out,nucl:nucleotide.out\n",
      "k-mer length             \t0\n",
      "Alphabet size            \taa:21,nucl:5\n",
      "Compositional bias       \t1\n",
      "Compositional bias       \t1\n",
      "Max sequence length      \t65535\n",
      "Max results per query    \t300\n",
      "Mask residues            \t1\n",
      "Mask residues probability\t0.9\n",
      "Mask lower case residues \t0\n",
      "Spaced k-mers            \t1\n",
      "Spaced k-mer pattern     \t\n",
      "Sensitivity              \t7.5\n",
      "k-score                  \tseq:0,prof:0\n",
      "Check compatible         \t0\n",
      "Search type              \t0\n",
      "Split database           \t0\n",
      "Split memory limit       \t0\n",
      "Verbosity                \t3\n",
      "Threads                  \t8\n",
      "Min codons in orf        \t30\n",
      "Max codons in length     \t32734\n",
      "Max orf gaps             \t2147483647\n",
      "Contig start mode        \t2\n",
      "Contig end mode          \t2\n",
      "Orf start mode           \t1\n",
      "Forward frames           \t1,2,3\n",
      "Reverse frames           \t1,2,3\n",
      "Translation table        \t1\n",
      "Translate orf            \t0\n",
      "Use all table starts     \tfalse\n",
      "Offset of numeric ids    \t0\n",
      "Create lookup            \t0\n",
      "Compressed               \t0\n",
      "Add orf stop             \tfalse\n",
      "Overlap between sequences\t0\n",
      "Sequence split mode      \t1\n",
      "Header split mode        \t0\n",
      "Strand selection         \t1\n",
      "Remove temporary files   \tfalse\n",
      "\n",
      "indexdb targetDB/targetDB targetDB/targetDB --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -k 0 --alph-size aa:21,nucl:5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-seq-len 65535 --max-seqs 300 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --spaced-kmer-mode 1 -s 7.5 --k-score seq:0,prof:0 --check-compatible 0 --search-type 0 --split 0 --split-memory-limit 0 -v 3 --threads 8 \n",
      "\n",
      "Estimated memory consumption: 2G\n",
      "Write VERSION (0)\n",
      "Write META (1)\n",
      "Write SCOREMATRIX3MER (4)\n",
      "Write SCOREMATRIX2MER (3)\n",
      "Write SCOREMATRIXNAME (2)\n",
      "Write SPACEDPATTERN (23)\n",
      "Write GENERATOR (22)\n",
      "Write DBR1INDEX (5)\n",
      "Write DBR1DATA (6)\n",
      "Write HDR1INDEX (18)\n",
      "Write HDR1DATA (19)\n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 570.16K 22s 408ms    \n",
      "Index table: Masked residues: 3242699\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 570.16K 20s 739ms    \n",
      "Index statistics\n",
      "Entries:          196141613\n",
      "DB size:          1610 MB\n",
      "Avg k-mer size:   3.064713\n",
      "Top 10 k-mers\n",
      "    GPGGTL\t1842\n",
      "    GQSWTV\t1705\n",
      "    WGMFAT\t1637\n",
      "    PGVFEV\t1637\n",
      "    VLWQFW\t1622\n",
      "    AYIRPN\t1586\n",
      "    RSPKGV\t1584\n",
      "    TPHKWY\t1559\n",
      "    HIRVGW\t1547\n",
      "    ITLSPY\t1540\n",
      "Write ENTRIES (9)\n",
      "Write ENTRIESOFFSETS (10)\n",
      "Write SEQINDEXDATASIZE (15)\n",
      "Write SEQINDEXSEQOFFSET (16)\n",
      "Write SEQINDEXDATA (14)\n",
      "Write ENTRIESNUM (12)\n",
      "Write SEQCOUNT (13)\n",
      "Time for merging to targetDB.idx: 0h 0m 0s 85ms\n",
      "Time for processing: 0h 0m 56s 236ms\n"
     ]
    }
   ],
   "source": [
    "#It is recommended to compute the index if the targetDB is reused for several searches.\n",
    "mmseqs createindex targetDB/targetDB tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6aa50538-59f9-43e5-823a-1f1ae2f342b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# MMseqs2 can produce a high IO on the file system. It is recommended to create a temporary folder on a local drive.\n",
    "# mkdir tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ec65524b-9f12-4e47-9dba-95960ca5485e",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir resultDB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e756341c-08c4-40a0-95b9-11fd07e978c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "search queryDB/queryDB targetDB/targetDB resultDB/resultDB tmp -a \n",
      "\n",
      "MMseqs Version:                        \t14.7e284\n",
      "Substitution matrix                    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Add backtrace                          \ttrue\n",
      "Alignment mode                         \t2\n",
      "Alignment mode                         \t0\n",
      "Allow wrapped scoring                  \tfalse\n",
      "E-value threshold                      \t0.001\n",
      "Seq. id. threshold                     \t0\n",
      "Min alignment length                   \t0\n",
      "Seq. id. mode                          \t0\n",
      "Alternative alignments                 \t0\n",
      "Coverage threshold                     \t0\n",
      "Coverage mode                          \t0\n",
      "Max sequence length                    \t65535\n",
      "Compositional bias                     \t1\n",
      "Compositional bias                     \t1\n",
      "Max reject                             \t2147483647\n",
      "Max accept                             \t2147483647\n",
      "Include identical seq. id.             \tfalse\n",
      "Preload mode                           \t0\n",
      "Pseudo count a                         \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                         \tsubstitution:4.100,context:5.800\n",
      "Score bias                             \t0\n",
      "Realign hits                           \tfalse\n",
      "Realign score bias                     \t-0.2\n",
      "Realign max seqs                       \t2147483647\n",
      "Correlation score weight               \t0\n",
      "Gap open cost                          \taa:11,nucl:5\n",
      "Gap extension cost                     \taa:1,nucl:2\n",
      "Zdrop                                  \t40\n",
      "Threads                                \t8\n",
      "Compressed                             \t0\n",
      "Verbosity                              \t3\n",
      "Seed substitution matrix               \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                            \t5.7\n",
      "k-mer length                           \t0\n",
      "k-score                                \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                          \taa:21,nucl:5\n",
      "Max results per query                  \t300\n",
      "Split database                         \t0\n",
      "Split mode                             \t2\n",
      "Split memory limit                     \t0\n",
      "Diagonal scoring                       \ttrue\n",
      "Exact k-mer matching                   \t0\n",
      "Mask residues                          \t1\n",
      "Mask residues probability              \t0.9\n",
      "Mask lower case residues               \t0\n",
      "Minimum diagonal score                 \t15\n",
      "Selected taxa                          \t\n",
      "Spaced k-mers                          \t1\n",
      "Spaced k-mer pattern                   \t\n",
      "Local temporary path                   \t\n",
      "Rescore mode                           \t0\n",
      "Remove hits by seq. id. and coverage   \tfalse\n",
      "Sort results                           \t0\n",
      "Mask profile                           \t1\n",
      "Profile E-value threshold              \t0.1\n",
      "Global sequence weighting              \tfalse\n",
      "Allow deletions                        \tfalse\n",
      "Filter MSA                             \t1\n",
      "Use filter only at N seqs              \t0\n",
      "Maximum seq. id. threshold             \t0.9\n",
      "Minimum seq. id.                       \t0.0\n",
      "Minimum score per column               \t-20\n",
      "Minimum coverage                       \t0\n",
      "Select N most diverse seqs             \t1000\n",
      "Pseudo count mode                      \t0\n",
      "Gap pseudo count                       \t10\n",
      "Min codons in orf                      \t30\n",
      "Max codons in length                   \t32734\n",
      "Max orf gaps                           \t2147483647\n",
      "Contig start mode                      \t2\n",
      "Contig end mode                        \t2\n",
      "Orf start mode                         \t1\n",
      "Forward frames                         \t1,2,3\n",
      "Reverse frames                         \t1,2,3\n",
      "Translation table                      \t1\n",
      "Translate orf                          \t0\n",
      "Use all table starts                   \tfalse\n",
      "Offset of numeric ids                  \t0\n",
      "Create lookup                          \t0\n",
      "Add orf stop                           \tfalse\n",
      "Overlap between sequences              \t0\n",
      "Sequence split mode                    \t1\n",
      "Header split mode                      \t0\n",
      "Chain overlapping alignments           \t0\n",
      "Merge query                            \t1\n",
      "Search type                            \t0\n",
      "Search iterations                      \t1\n",
      "Start sensitivity                      \t4\n",
      "Search steps                           \t1\n",
      "Exhaustive search mode                 \tfalse\n",
      "Filter results during exhaustive search\t0\n",
      "Strand selection                       \t1\n",
      "LCA search mode                        \tfalse\n",
      "Disk space limit                       \t0\n",
      "MPI runner                             \t\n",
      "Force restart with latest tmp          \tfalse\n",
      "Remove temporary files                 \tfalse\n",
      "\n",
      "prefilter queryDB/queryDB targetDB/targetDB.idx tmp/2216815330233090478/pref_0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 300 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 -s 5.7 \n",
      "\n",
      "Index version: 16\n",
      "Generated by:  14.7e284\n",
      "ScoreMatrix:  VTML80.out\n",
      "Query database size: 1 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 570157 type: Aminoacid\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 112\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1\n",
      "Target db start 1 to 570157\n",
      "[=================================================================] 100.00% 1 eta -\n",
      "\n",
      "453.000000 k-mers per position\n",
      "9873 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "74 sequences passed prefiltering per query sequence\n",
      "74 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 654ms\n",
      "align queryDB/queryDB targetDB/targetDB.idx tmp/2216815330233090478/pref_0 resultDB/resultDB --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 1 --alignment-mode 2 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Index version: 16\n",
      "Generated by:  14.7e284\n",
      "ScoreMatrix:  VTML80.out\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1 type: Aminoacid\n",
      "Target database size: 570157 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 1 eta -\n",
      "Time for merging to resultDB: 0h 0m 0s 0ms\n",
      "74 alignments calculated\n",
      "1 sequence pairs passed the thresholds (0.013514 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 609ms\n"
     ]
    }
   ],
   "source": [
    "mmseqs search queryDB/queryDB targetDB/targetDB resultDB/resultDB tmp -a \n",
    "\n",
    "#-format-output \"query,target,qaln,taln,alnlen,evalue,nident,pident,bits\" not working?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "f687daba-93e7-40e7-b7af-ec9938e6dd2f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "convertalis queryDB/queryDB targetDB/targetDB resultDB/resultDB resultDB/resultDB.m8 \n",
      "\n",
      "MMseqs Version:        \t14.7e284\n",
      "Substitution matrix    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Alignment format       \t0\n",
      "Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Translation table      \t1\n",
      "Gap open cost          \taa:11,nucl:5\n",
      "Gap extension cost     \taa:1,nucl:2\n",
      "Database output        \tfalse\n",
      "Preload mode           \t0\n",
      "Search type            \t0\n",
      "Threads                \t8\n",
      "Compressed             \t0\n",
      "Verbosity              \t3\n",
      "\n",
      "[=================================================================] 100.00% 1 eta -\n",
      "Time for merging to resultDB.m8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 237ms\n"
     ]
    }
   ],
   "source": [
    "mmseqs convertalis queryDB/queryDB targetDB/targetDB resultDB/resultDB resultDB/resultDB.m8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "6bd93766-680e-497d-88bd-1124cfc51300",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A0A0C5B5G6\tA0A0C5B5G6\t1.000\t16\t0\t0\t1\t16\t1\t16\t3.891E-05\t40\n"
     ]
    }
   ],
   "source": [
    "head resultDB/resultDB.m8 \n",
    "# (1,2) identifiers for query and target sequences/profiles, (3) sequence identity, (4) alignment length, \n",
    "# (5) number of mismatches, (6) number of gap openings, (7-8, 9-10) domain start and end-position in query and in target, \n",
    "# (11) E-value, and (12) bit score."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f431d984-6ddd-4f46-b42d-7dcac2cb369c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "488f597f-560b-48d2-b76f-7a71c78a6867",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af0d1432-3451-44f7-b6c9-e0a22d23a6a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8333005d-c994-42b6-8662-77a7bc0d05fe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35324e1a-a3de-4464-a928-ec354d321038",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa9f6eb0-8808-4d9b-8f33-9f98145a4d34",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3ff0b3d-d782-465f-91c2-8802d32c1419",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
